---
title: "Mutation frequencies"
author: "Iana Amke, Kara Tsang, Kat Holt"
date: "Last compiled on `r format(Sys.time(), '%d/%m/%y')"
output: 
  html_document: 
    df_print: paged
    fig_width: 9
    fig_height: 8
    mode: selfcontained
    highlight: pygments
    theme: 
      bootswatch: flatly
      code_font:
        google: "JetBrains Mono"
editor_options: 
  chunk_output_type: inline
---

```{=html}
<style type="text/css">
.main-container {
  max-width: 1600px !important;
  margin-left: auto;
  margin-right: auto;
}
</style>
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = 'png', echo = TRUE, message = FALSE, warning = FALSE,
                      cache = FALSE, fig.align = 'center')
knitr::opts_knit$set(root.dir = getwd())

# Data wrangling
library(tidyverse)
library(dplyr)
library(countrycode)
library(pheatmap)
library(ggpubr)
library(ggiraphExtra)
library(ggupset)
library(scales)
library(reshape2)
library(ComplexUpset)
library(RColorBrewer)
library(naniar)
library(grid)
library(gtsummary)
library(broom)
library(patchwork)
library(ggrepel)
library(magrittr)
library(epiR)

# https://krassowski.github.io/complex-upset/articles/Examples_R.html
```

# load data
``` {r}
geno <- read_delim("tables/TableS1_genotypes.tsv")

pheno <- read_delim("tables/TableS2_phenotypes_metadata.tsv")

cipro_antibiogram <- full_join(geno,pheno)

```


# Summarise mutations - Table 1, Table S5

## long form, one row per strain/determinant combination
``` {r}
cipro_res_mech <- cipro_antibiogram %>%
  pivot_longer(c(Flq_mutations,Flq_acquired), names_to = "Flq.resistance.mech", 
               values_to = "Resistance.mech.type") %>%
  separate(Resistance.mech.type, into = c("type_A", "type_B", "type_C", "type_D", "type_E", "type_F", "type_G", "type_H", "type_I", "type_J"), sep = "[;]")

cipro_res_mech <- cipro_res_mech %>%
  pivot_longer(cols = type_A:type_J, values_to = "Resistance.mech.type") %>%
  drop_na(Resistance.mech.type) %>%
  select(-name) %>%
  distinct()

cipro_res_mech <- cipro_res_mech %>% 
  bind_rows( # add aac6 as its own mechanism
      cipro_antibiogram %>% 
        filter(aac6 == 1) %>% 
        mutate(Flq.resistance.mech = "Aac6_gene", Resistance.mech.type="aac(6')-Ib-cr")
      ) %>%
  bind_rows( # add null rows for aac6 as its own mechanism
      cipro_antibiogram %>% 
        filter(aac6 == 0) %>% 
        mutate(Flq.resistance.mech = "Aac6_gene", Resistance.mech.type="-")
      )

# check that we have the expected number of entries for each mechanism
cipro_res_mech %>% filter(Flq.resistance.mech=="Flq_mutations") %>% select(strain) %>% unique() %>% nrow()
cipro_res_mech %>% filter(Flq.resistance.mech=="Flq_acquired") %>% select(strain) %>% unique() %>% nrow()
cipro_res_mech %>% filter(Flq.resistance.mech=="Aac6_gene") %>% select(strain) %>% unique() %>% nrow()


```

# Numbers for text - para 2
``` {r}
# total isolates
totalN <- cipro_antibiogram %>% select(strain) %>% unique() %>% nrow()
totalN

# number of isolates with each type of mutation
total_withQRDR <- cipro_res_mech %>% filter(Flq.resistance.mech=="Flq_mutations") %>% filter(Resistance.mech.type!="-") %>% select(strain) %>% unique() %>% nrow()
total_withQRDR
total_withQRDR/totalN*100

total_withPMWR <- cipro_res_mech %>% filter(Flq.resistance.mech=="Flq_acquired") %>% filter(Resistance.mech.type!="-") %>% select(strain) %>% unique() %>% nrow()
total_withPMWR
total_withPMWR/totalN*100

total_withAAC <- cipro_res_mech %>% filter(Flq.resistance.mech=="Aac6_gene") %>% filter(Resistance.mech.type!="-") %>% select(strain) %>% unique() %>% nrow()
total_withAAC
total_withAAC/totalN*100

# freq of determinants in susceptible strains
totalS <- cipro_res_mech %>% filter(Resistance.phenotype=="S") %>% select(strain) %>% unique() %>% nrow()

S_withQRDR <- cipro_res_mech %>% filter(Flq.resistance.mech=="Flq_mutations") %>% filter(Resistance.mech.type!="-") %>% filter(Resistance.phenotype=="S") %>% select(strain) %>% unique() %>% nrow()
S_withQRDR
S_withQRDR/totalS*100

S_withPMQR <- cipro_res_mech %>% filter(Flq.resistance.mech=="Flq_acquired") %>% filter(Resistance.mech.type!="-") %>% filter(Resistance.phenotype=="S") %>% select(strain) %>% unique() %>% nrow()
S_withPMQR
S_withPMQR/totalS*100

S_withAac <- cipro_res_mech %>% filter(Flq.resistance.mech=="Aac6_gene") %>% filter(Resistance.mech.type!="-") %>% filter(Resistance.phenotype=="S") %>% select(strain) %>% unique() %>% nrow()
S_withAac
S_withAac/totalS*100

# freq of determinants in susceptible strains
totalR <- cipro_res_mech %>% filter(Resistance.phenotype=="R") %>% select(strain) %>% unique() %>% nrow()

R_withQRDR <- cipro_res_mech %>% filter(Flq.resistance.mech=="Flq_mutations") %>% filter(Resistance.mech.type!="-") %>% filter(Resistance.phenotype=="R") %>% select(strain) %>% unique() %>% nrow()
R_withQRDR
R_withQRDR/totalR*100

R_withPMQR <- cipro_res_mech %>% filter(Flq.resistance.mech=="Flq_acquired") %>% filter(Resistance.mech.type!="-") %>% filter(Resistance.phenotype=="R") %>% select(strain) %>% unique() %>% nrow()
R_withPMQR
R_withPMQR/totalR*100

R_withAac <- cipro_res_mech %>% filter(Flq.resistance.mech=="Aac6_gene") %>% filter(Resistance.mech.type!="-") %>% filter(Resistance.phenotype=="R") %>% select(strain) %>% unique() %>% nrow()
R_withAac
R_withAac/totalR*100
```


# Table S5 - freq per determinant
``` {r TableS5}
freq_per_det <- cipro_res_mech %>% 
  select(strain, Resistance.mech.type) %>%
  mutate(Resistance.mech.type=gsub("[\\^\\*\\?]", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v1", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v2", "", Resistance.mech.type)) %>%
  group_by(Resistance.mech.type) %>% 
  count() %>% rename(n_gene=n) %>%
  bind_rows( 
    cipro_antibiogram %>% 
    select(`GyrA-83`:`ParC-84`) %>% 
    apply(2, sum) %>% 
    as_tibble(rownames="Resistance.mech.type") %>%
    rename(n_gene=value)
  )

freq_per_pheno <- cipro_antibiogram %>% 
  group_by(Resistance.phenotype) %>% 
  count() %>% rename(n_pheno=n)

freq_per_det_byPheno <- cipro_res_mech %>% 
  mutate(Resistance.mech.type=gsub("[\\^\\*\\?]", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v1", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v2", "", Resistance.mech.type)) %>%
  group_by(Flq.resistance.mech, Resistance.mech.type, Resistance.phenotype) %>% 
  count() %>% 
  bind_rows( # add `GyrA-83`
    cipro_antibiogram %>% 
      filter(`GyrA-83`>0) %>%
      group_by(Resistance.phenotype) %>% 
      count() %>%
      mutate(Resistance.mech.type="GyrA-83")
  ) %>%
  bind_rows( # add `GyrA-87`
    cipro_antibiogram %>% 
      filter(`GyrA-87`>0) %>%
      group_by(Resistance.phenotype) %>% 
      count() %>%
      mutate(Resistance.mech.type="GyrA-87")
  ) %>%
  bind_rows( # add `ParC-80`
    cipro_antibiogram %>% 
      filter(`ParC-80`>0) %>%
      group_by(Resistance.phenotype) %>% 
      count() %>%
      mutate(Resistance.mech.type="ParC-80")
  ) %>%
  bind_rows( # add `ParC-84`
    cipro_antibiogram %>% 
      filter(`ParC-84`>0) %>%
      group_by(Resistance.phenotype) %>% 
      count() %>%
      mutate(Resistance.mech.type="ParC-84")
  ) %>%
  left_join(freq_per_det) %>% 
  left_join(freq_per_pheno) %>% 
  filter(Resistance.mech.type != "-") %>%
  mutate(freq_within_phenoGroup = n/n_pheno) %>% 
#  mutate(prop_test_p = 
#           prop.test(c(n,n_gene-n), 
#                     c(n_pheno, nrow(cipro_antibiogram) - n_pheno))$p.value) %>%
  mutate(G1P1=n) %>% 
  mutate(G1P0=n_gene-n) %>%
  mutate(G0P1=n_pheno-n) %>%
  mutate(G0P0=nrow(cipro_antibiogram) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  select(Resistance.mech.type, Resistance.phenotype, n, freq_within_phenoGroup, OR, p_val) %>%
  mutate(summary=paste0("n=", n, " (", round(freq_within_phenoGroup*100,2), "%)"))


freq_per_phenoWT <- cipro_antibiogram %>% 
  group_by(WT_vs_nonWT) %>% 
  count() %>% rename(n_pheno=n)

freq_per_det_byWT <- cipro_res_mech %>% 
  mutate(Resistance.mech.type=gsub("[\\^\\*\\?]", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v1", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v2", "", Resistance.mech.type)) %>%
  group_by(Flq.resistance.mech, Resistance.mech.type, WT_vs_nonWT) %>% 
  count() %>% 
  bind_rows( # add `GyrA-83`
    cipro_antibiogram %>% 
      filter(`GyrA-83`>0) %>%
      group_by(WT_vs_nonWT) %>% 
      count() %>%
      mutate(Resistance.mech.type="GyrA-83")
  ) %>%
  bind_rows( # add `GyrA-87`
    cipro_antibiogram %>% 
      filter(`GyrA-87`>0) %>%
      group_by(WT_vs_nonWT) %>% 
      count() %>%
      mutate(Resistance.mech.type="GyrA-87")
  ) %>%
  bind_rows( # add `ParC-80`
    cipro_antibiogram %>% 
      filter(`ParC-80`>0) %>%
      group_by(WT_vs_nonWT) %>% 
      count() %>%
      mutate(Resistance.mech.type="ParC-80")
  ) %>%
  bind_rows( # add `ParC-84`
    cipro_antibiogram %>% 
      filter(`ParC-84`>0) %>%
      group_by(WT_vs_nonWT) %>% 
      count() %>%
      mutate(Resistance.mech.type="ParC-84")
  ) %>%
  left_join(freq_per_det) %>% 
  left_join(freq_per_phenoWT) %>% 
  filter(Resistance.mech.type != "-") %>%
  mutate(freq_within_phenoGroup = n/n_pheno) %>% 
#  mutate(prop_test_p = 
#           prop.test(c(n,n_gene-n), 
#                     c(n_pheno, nrow(cipro_antibiogram) - n_pheno))$p.value) %>%
  mutate(G1P1=n) %>% 
  mutate(G1P0=n_gene-n) %>%
  mutate(G0P1=n_pheno-n) %>%
  mutate(G0P0=nrow(cipro_antibiogram) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  select(Resistance.mech.type, WT_vs_nonWT, n, freq_within_phenoGroup, OR, p_val) %>%
  mutate(summary=paste0("n=", n, " (", round(freq_within_phenoGroup*100,2), "%)"))

```

``` {r}
# assemble summary table
freq_per_det_byPheno_summary <- freq_per_det_byPheno %>%
  filter(Resistance.phenotype=="R") %>% # add resistance columns
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(Resistant=summary) %>%
  rename(R_pval=p_val) %>%
  rename(R_OR=OR)

freq_per_det_byPheno_summary <- freq_per_det_byWT %>%
  filter(WT_vs_nonWT=="non-wt (I/R)") %>% # add NWT column
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(NWT=summary) %>% 
  full_join(freq_per_det_byPheno_summary) %>%
  rename(NWT_pval=p_val) %>%
  rename(NWT_OR=OR)

freq_per_det_byPheno_summary <- freq_per_det_byPheno %>%
  filter(Resistance.phenotype=="S") %>% # add susceptible/WT column
  select(Resistance.mech.type, summary) %>%
  rename(`Susceptible`=summary) %>% 
  full_join(freq_per_det_byPheno_summary) %>%
  full_join(freq_per_det) %>% # add totals column
  filter(Resistance.mech.type!="-") %>%
  mutate(Total=paste0("n=", n_gene, " (", round(n_gene/nrow(cipro_antibiogram)*100,2), "%)")) %>% 
  #arrange(-n_gene) %>% # sort by frequency of determinant
  arrange(Resistance.mech.type) %>% # sort alphabetically
  select(-n_gene) %>%
  rename(Determinant=Resistance.mech.type) %>%
  relocate(Total, .after=Determinant) %>%
  mutate(R_test=if_else(R_OR!="Inf",paste0(round(R_OR,2), " (p=",scientific(R_pval,2), ")"), NA)) %>%
  mutate(NWT_test=if_else(NWT_OR!="Inf",paste0(round(NWT_OR,2), " (p=",scientific(NWT_pval,2), ")"), NA)) %>%
  select(Determinant, Total, Susceptible, Resistant, R_test, NWT, NWT_test)

```

# write out Table 5
``` {r}
freq_per_det_byPheno_summary %>%
  mutate(Susceptible=if_else(is.na(Susceptible), "0", Susceptible)) %>%
  mutate(NWT=if_else(is.na(NWT), "0", NWT)) %>%
  mutate(Resistant=if_else(is.na(Resistant), "0", Resistant)) %>%
  mutate(R_test=if_else(is.na(R_test), "-", R_test)) %>%
  mutate(NWT_test=if_else(is.na(NWT_test), "-", NWT_test)) %>%
  rename(`Total (% of genomes)`=Total) %>%
  rename(`Association with Resistance (OR, p-value)`=R_test) %>%
  rename(`Association with NWT (OR, p-value)`=NWT_test) %>%
  rename(`Resistant (% of phenotype)`=Resistant) %>%
  rename(`Susceptible (% of phenotype)`=Susceptible) %>%
  rename(`NWT (% of phenotype)`=NWT) %>%
write_csv(file="tables/TableS5_freq_per_det_byPheno_summary.csv")
```


# Table 1
## freq stats for single QRDR mutations found in genomes without acquired genes or aac
``` {r Table1}
strains_singleQRDR_noAcquired <- cipro_antibiogram %>%
  filter(QRDR_mutations==1 & aac6==0 & acquired_genes==0) %>%
  pull(strain)

strains_01QRDR_noAcquired <- cipro_antibiogram %>%
  filter(QRDR_mutations<=1 & aac6==0 & acquired_genes==0) %>%
  pull(strain)

freq_per_det_01QRDR_noAcquired <- cipro_res_mech %>% 
  filter(strain %in% strains_01QRDR_noAcquired) %>%
  select(strain, Resistance.mech.type) %>%
  group_by(Resistance.mech.type) %>% 
  count() %>% 
  rename(n_gene=n) %>% 
  filter(Resistance.mech.type!="-") %>%
  bind_rows( 
    cipro_antibiogram %>% 
    filter(strain %in% strains_01QRDR_noAcquired) %>%
    select(`GyrA-83`:`ParC-84`) %>% 
    apply(2, sum) %>% 
    as_tibble(rownames="Resistance.mech.type") %>%
    rename(n_gene=value)
  )

freq_per_pheno_singleQRDR_noAcquired <- cipro_antibiogram %>% 
  filter(strain %in% strains_01QRDR_noAcquired) %>%
  group_by(Resistance.phenotype) %>% 
  count() %>% rename(n_pheno=n)
  
freq_per_det_byPheno_singleQRDRonly <- cipro_res_mech %>% 
  filter(strain %in% strains_singleQRDR_noAcquired) %>%
  filter(Flq.resistance.mech=="Flq_mutations") %>%
  group_by(Flq.resistance.mech, Resistance.mech.type, Resistance.phenotype) %>% 
  count() %>% 
  bind_rows( # add `GyrA-83`
    cipro_antibiogram %>% 
      filter(strain %in% strains_singleQRDR_noAcquired) %>%
      filter(`GyrA-83`>0) %>%
      group_by(Resistance.phenotype) %>% 
      count() %>%
      mutate(Resistance.mech.type="GyrA-83")
  ) %>%
  bind_rows( # add `GyrA-87`
    cipro_antibiogram %>% 
      filter(strain %in% strains_singleQRDR_noAcquired) %>%
      filter(`GyrA-87`>0) %>%
      group_by(Resistance.phenotype) %>% 
      count() %>%
      mutate(Resistance.mech.type="GyrA-87")
  ) %>%
  bind_rows( # add `ParC-80`
    cipro_antibiogram %>% 
      filter(strain %in% strains_singleQRDR_noAcquired) %>%
      filter(`ParC-80`>0) %>%
      group_by(Resistance.phenotype) %>% 
      count() %>%
      mutate(Resistance.mech.type="ParC-80")
  ) %>%
  bind_rows( # add `ParC-84`
    cipro_antibiogram %>% 
      filter(strain %in% strains_singleQRDR_noAcquired) %>%
      filter(`ParC-84`>0) %>%
      group_by(Resistance.phenotype) %>% 
      count() %>%
      mutate(Resistance.mech.type="ParC-84")
  ) %>%
  left_join(freq_per_det_01QRDR_noAcquired) %>% 
  left_join(freq_per_pheno_singleQRDR_noAcquired) %>% 
  filter(Resistance.mech.type != "-") %>%
 # mutate(freq_within_phenoGroup = n/n_pheno) %>% 
  mutate(G1P1=n) %>% 
  mutate(G1P0=n_gene-n) %>%
  mutate(G0P1=n_pheno-n) %>%
  mutate(G0P0=length(strains_01QRDR_noAcquired) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  mutate(summary=paste0("n=", n, " (", round(n/n_gene*100,2), "%)")) %>%
  select(Resistance.mech.type, Resistance.phenotype, n, OR, p_val, summary, G0P0, G1P0, G0P1, G1P1) 

# NWT
freq_per_pheno_singleQRDR_noAcquired_NWT <- cipro_antibiogram %>% 
  filter(strain %in% strains_01QRDR_noAcquired) %>%
  group_by(WT_vs_nonWT) %>% 
  count() %>% rename(n_pheno=n)
  
freq_per_det_byPheno_singleQRDRonly_NWT <- cipro_res_mech %>% 
  filter(strain %in% strains_singleQRDR_noAcquired) %>%
  filter(Flq.resistance.mech=="Flq_mutations") %>%
  group_by(Flq.resistance.mech, Resistance.mech.type, WT_vs_nonWT) %>% 
  count() %>% 
  bind_rows( # add `GyrA-83`
    cipro_antibiogram %>% 
      filter(strain %in% strains_singleQRDR_noAcquired) %>%
      filter(`GyrA-83`>0) %>%
      group_by(WT_vs_nonWT) %>% 
      count() %>%
      mutate(Resistance.mech.type="GyrA-83")
  ) %>%
  bind_rows( # add `GyrA-87`
    cipro_antibiogram %>% 
      filter(strain %in% strains_singleQRDR_noAcquired) %>%
      filter(`GyrA-87`>0) %>%
      group_by(WT_vs_nonWT) %>% 
      count() %>%
      mutate(Resistance.mech.type="GyrA-87")
  ) %>%
  bind_rows( # add `ParC-80`
    cipro_antibiogram %>% 
      filter(strain %in% strains_singleQRDR_noAcquired) %>%
      filter(`ParC-80`>0) %>%
      group_by(WT_vs_nonWT) %>% 
      count() %>%
      mutate(Resistance.mech.type="ParC-80")
  ) %>%
  bind_rows( # add `ParC-84`
    cipro_antibiogram %>% 
      filter(strain %in% strains_singleQRDR_noAcquired) %>%
      filter(`ParC-84`>0) %>%
      group_by(WT_vs_nonWT) %>% 
      count() %>%
      mutate(Resistance.mech.type="ParC-84")
  ) %>%
  left_join(freq_per_det_01QRDR_noAcquired) %>% 
  left_join(freq_per_pheno_singleQRDR_noAcquired_NWT) %>% 
  filter(Resistance.mech.type != "-") %>%
  mutate(G1P1=n) %>% 
  mutate(G1P0=n_gene-n) %>%
  mutate(G0P1=n_pheno-n) %>%
  mutate(G0P0=length(strains_01QRDR_noAcquired) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  mutate(summary=paste0("n=", n, " (", round(n/n_gene*100,2), "%)")) %>%
  select(Resistance.mech.type, WT_vs_nonWT, n, OR, p_val, summary, G0P0, G1P0, G0P1, G1P1) 

# assemble summary table
freq_per_det_byPheno_summary_singleQRDR <- freq_per_det_byPheno_singleQRDRonly %>%
  filter(Resistance.phenotype=="R") %>% # add resistance columns
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(Resistant=summary) %>%
  rename(R_OR=OR) %>%
  rename(R_pval=p_val)

freq_per_det_byPheno_summary_singleQRDR <- freq_per_det_byPheno_singleQRDRonly_NWT %>%
  filter(WT_vs_nonWT=="non-wt (I/R)") %>% # add intermediate column
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(NWT=summary) %>% 
  full_join(freq_per_det_byPheno_summary_singleQRDR)%>%
  rename(NWT_OR=OR) %>%
  rename(NWT_pval=p_val)

freq_per_det_byPheno_summary_singleQRDR <- freq_per_det_byPheno_singleQRDRonly %>%
  filter(Resistance.phenotype=="S") %>% # add susceptible column
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(Susceptible=summary) %>% 
  full_join(freq_per_det_byPheno_summary_singleQRDR) %>%
  full_join(freq_per_det_01QRDR_noAcquired) %>% # add totals column
  filter(Resistance.mech.type!="-") %>%
  rename(Total=n_gene) %>% 
  arrange(Resistance.mech.type) %>% # sort by frequency of determinant
  rename(`Solo determinant`=Resistance.mech.type) %>%
  mutate(R_test=if_else(R_OR!="Inf",paste0(round(R_OR,2), " (p=",scientific(R_pval,2), ")"), NA)) %>%
  mutate(NWT_test=if_else(NWT_OR!="Inf",paste0(round(NWT_OR,2), " (p=",scientific(NWT_pval,2), ")"), NA)) %>%
  select(`Solo determinant`, Total, Susceptible, NWT, Resistant, R_test, NWT_test)

```

## freq stats for acquired genes found in genomes without QRDR or aac
``` {r }
strains_single_acquired_gene <- cipro_antibiogram %>%
  filter(QRDR_mutations==0 & aac6==0 & acquired_genes==1) %>%
  pull(strain)

strains_noQRDR_noAac <- cipro_antibiogram %>%
  filter(QRDR_mutations==0 & aac6==0) %>%
  pull(strain)

freq_per_det_noQRDR_noAac <- cipro_res_mech %>% 
  filter(strain %in% strains_noQRDR_noAac) %>%
  select(strain, Resistance.mech.type) %>%
  mutate(Resistance.mech.type=gsub("[\\^\\*\\?]", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v1", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v2", "", Resistance.mech.type)) %>%
  group_by(Resistance.mech.type) %>% 
  count() %>% 
  rename(n_gene=n) %>% 
  filter(Resistance.mech.type!="-")

freq_per_pheno_noQRDR_noAac <- cipro_antibiogram %>% 
  filter(strain %in% strains_noQRDR_noAac) %>%
  group_by(Resistance.phenotype) %>% 
  count() %>% rename(n_pheno=n)
  
freq_per_det_byPheno_acquiredGeneOnly <- cipro_res_mech %>% 
  filter(strain %in% strains_single_acquired_gene) %>%
  mutate(Resistance.mech.type=gsub("[\\^\\*\\?]", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v1", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v2", "", Resistance.mech.type)) %>%
  filter(Flq.resistance.mech=="Flq_acquired") %>%
  group_by(Flq.resistance.mech, Resistance.mech.type, Resistance.phenotype) %>% 
  count() %>% 
  left_join(freq_per_det_noQRDR_noAac) %>% 
  left_join(freq_per_pheno_noQRDR_noAac) %>% 
  filter(Resistance.mech.type != "-") %>%
  mutate(G1P1=n) %>% 
  mutate(G1P0=n_gene-n) %>%
  mutate(G0P1=n_pheno-n) %>%
  mutate(G0P0=length(strains_noQRDR_noAac) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  mutate(summary=paste0("n=", n, " (", round(n/n_gene*100,2), "%)")) %>%
  select(Resistance.mech.type, Resistance.phenotype, n, OR, p_val, summary, G0P0, G1P0, G0P1, G1P1) 


freq_per_pheno_noQRDR_noAac_NWT <- cipro_antibiogram %>% 
  filter(strain %in% strains_noQRDR_noAac) %>%
  group_by(WT_vs_nonWT) %>% 
  count() %>% rename(n_pheno=n)
  
freq_per_det_byPheno_acquiredGeneOnly_NWT <- cipro_res_mech %>% 
  filter(strain %in% strains_single_acquired_gene) %>%
  mutate(Resistance.mech.type=gsub("[\\^\\*\\?]", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v1", "", Resistance.mech.type)) %>%
  mutate(Resistance.mech.type=gsub(".v2", "", Resistance.mech.type)) %>%
  filter(Flq.resistance.mech=="Flq_acquired") %>%
  group_by(Flq.resistance.mech, Resistance.mech.type, WT_vs_nonWT) %>% 
  count() %>% 
  left_join(freq_per_det_noQRDR_noAac) %>% 
  left_join(freq_per_pheno_noQRDR_noAac_NWT) %>% 
  filter(Resistance.mech.type != "-") %>%
  mutate(G1P1=n) %>% 
  mutate(G1P0=n_gene-n) %>%
  mutate(G0P1=n_pheno-n) %>%
  mutate(G0P0=length(strains_noQRDR_noAac) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  mutate(summary=paste0("n=", n, " (", round(n/n_gene*100,2), "%)")) %>%
  select(Resistance.mech.type, WT_vs_nonWT, n, OR, p_val, summary, G0P0, G1P0, G0P1, G1P1) 

# assemble summary table
freq_per_det_byPheno_summary_singleAcquired <- freq_per_det_byPheno_acquiredGeneOnly %>%
  filter(Resistance.phenotype=="R") %>% # add resistance columns
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(Resistant=summary) %>%
  rename(R_OR=OR) %>%
  rename(R_pval=p_val)

freq_per_det_byPheno_summary_singleAcquired <- freq_per_det_byPheno_acquiredGeneOnly_NWT %>%
  filter(WT_vs_nonWT=="non-wt (I/R)") %>% # add intermediate column
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(NWT=summary) %>% 
  rename(NWT_OR=OR) %>%
  rename(NWT_pval=p_val) %>%
  full_join(freq_per_det_byPheno_summary_singleAcquired)

freq_per_det_byPheno_summary_singleAcquired <- freq_per_det_byPheno_acquiredGeneOnly %>%
  filter(Resistance.phenotype=="S") %>% # add susceptible column
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(Susceptible=summary) %>% 
  full_join(freq_per_det_byPheno_summary_singleAcquired) %>%
  full_join(freq_per_det_noQRDR_noAac) %>% # add totals column
  filter(Resistance.mech.type!="-") %>%
  rename(Total=n_gene) %>% 
  arrange(Resistance.mech.type) %>% # sort by frequency of determinant
  rename(`Solo determinant`=Resistance.mech.type) %>%
  mutate(R_test=if_else(R_OR!="Inf",paste0(round(R_OR,2), " (p=",scientific(R_pval,2), ")"), NA)) %>%
  mutate(NWT_test=if_else(NWT_OR!="Inf",paste0(round(NWT_OR,2), " (p=",scientific(NWT_pval,2), ")"), NA)) %>%
  select(`Solo determinant`, Total, Susceptible, NWT, Resistant, R_test, NWT_test)



```


#### aac6 (in absence of QRDR/qnr/qep, compare to denominator of all lacking QRDR/qnr/qep)

``` {r}
strains_aac6_noQRDRqnr <- cipro_antibiogram %>%
  filter(QRDR_mutations==0 & aac6==1 & acquired_genes==0) %>%
  pull(strain)

strains_noQRDRqnr <- cipro_antibiogram %>%
  filter(QRDR_mutations==0 & acquired_genes==0) %>%
  pull(strain)

freq_per_pheno_noQRDRqnr <- cipro_antibiogram %>% 
  filter(strain %in% strains_noQRDRqnr) %>%
  group_by(Resistance.phenotype) %>% 
  count() %>% rename(n_pheno=n)

freq_per_pheno_noQRDRqnr_NWT <- cipro_antibiogram %>% 
  filter(strain %in% strains_noQRDRqnr) %>%
  group_by(WT_vs_nonWT) %>% 
  count() %>% rename(n_pheno=n)

freq_per_det_noQRDRqnr <- cipro_res_mech %>% 
  filter(strain %in% strains_noQRDRqnr) %>%
  select(strain, Resistance.mech.type) %>%
  group_by(Resistance.mech.type) %>% 
  count() %>% 
  rename(n_gene=n) %>% 
  filter(Resistance.mech.type!="-")
  
freq_per_det_byPheno_noQRDRqnr <- cipro_res_mech %>% 
  filter(strain %in% strains_noQRDRqnr) %>%
  filter(Flq.resistance.mech=="Aac6_gene") %>%
  group_by(Flq.resistance.mech, Resistance.mech.type, Resistance.phenotype) %>% 
  count() %>%
  left_join(freq_per_det_noQRDRqnr) %>% 
  left_join(freq_per_pheno_noQRDRqnr) %>% 
  filter(Resistance.mech.type != "-") %>%
  mutate(G1P1=n) %>% 
  mutate(G1P0=n_gene-n) %>%
  mutate(G0P1=n_pheno-n) %>%
  mutate(G0P0=length(strains_noQRDR_noAac) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  mutate(summary=paste0("n=", n, " (", round(n/n_gene*100,2), "%)")) %>%
  select(Resistance.mech.type, Resistance.phenotype, n, OR, p_val, summary, G0P0, G1P0, G0P1, G1P1) 

freq_per_det_byPheno_noQRDRqnr_NWT <- cipro_res_mech %>% 
  filter(strain %in% strains_noQRDRqnr) %>%
  filter(Flq.resistance.mech=="Aac6_gene") %>%
  group_by(Flq.resistance.mech, Resistance.mech.type, WT_vs_nonWT) %>% 
  count() %>%
  left_join(freq_per_det_noQRDRqnr) %>% 
  left_join(freq_per_pheno_noQRDRqnr_NWT) %>% 
  filter(Resistance.mech.type != "-") %>%
  mutate(G1P1=n) %>% 
  mutate(G1P0=n_gene-n) %>%
  mutate(G0P1=n_pheno-n) %>%
  mutate(G0P0=length(strains_noQRDR_noAac) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  mutate(summary=paste0("n=", n, " (", round(n/n_gene*100,2), "%)")) %>%
  select(Resistance.mech.type, WT_vs_nonWT, n, OR, p_val, summary, G0P0, G1P0, G0P1, G1P1) 

# assemble summary table
freq_per_det_byPheno_summary_aac <- freq_per_det_byPheno_noQRDRqnr %>%
  filter(Resistance.phenotype=="R") %>% # add resistance columns
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(Resistant=summary) %>%
  rename(R_OR=OR) %>%
  rename(R_pval=p_val)

freq_per_det_byPheno_summary_aac <- freq_per_det_byPheno_noQRDRqnr_NWT %>%
  filter(WT_vs_nonWT=="non-wt (I/R)") %>% # add intermediate column
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(NWT=summary) %>% 
  rename(NWT_OR=OR) %>%
  rename(NWT_pval=p_val) %>%
  full_join(freq_per_det_byPheno_summary_aac)

freq_per_det_byPheno_summary_aac <- freq_per_det_byPheno_noQRDRqnr %>%
  filter(Resistance.phenotype=="S") %>% # add susceptible column
  select(Resistance.mech.type, summary, OR, p_val) %>%
  rename(Susceptible=summary) %>% 
  full_join(freq_per_det_byPheno_summary_aac) %>%
  full_join(freq_per_det_noQRDRqnr) %>% # add totals column
  filter(Resistance.mech.type!="-") %>%
  rename(Total=n_gene) %>% 
  arrange(Resistance.mech.type) %>% # sort by frequency of determinant
  rename(`Solo determinant`=Resistance.mech.type) %>%
  mutate(R_test=if_else(R_OR!="Inf",paste0(round(R_OR,2), " (p=",scientific(R_pval,2), ")"), NA)) %>%
  mutate(NWT_test=if_else(NWT_OR!="Inf",paste0(round(NWT_OR,2), " (p=",scientific(NWT_pval,2), ")"), NA)) %>%
  select(`Solo determinant`, Total, Susceptible, NWT, Resistant, R_test, NWT_test)
```


## write out Table 1 - solo determinants and association with phenotype
``` {r}

bind_rows(freq_per_det_byPheno_summary_singleQRDR, 
          freq_per_det_byPheno_summary_singleAcquired, 
          freq_per_det_byPheno_summary_aac) %>% 
  rename(R_OR=R_test, NWT_OR=NWT_test) %>%
  relocate(NWT, .before=NWT_OR) %>%
  mutate(Susceptible=if_else(is.na(Susceptible), "0", Susceptible)) %>%
  mutate(NWT=if_else(is.na(NWT), "0", NWT)) %>%
  mutate(Resistant=if_else(is.na(Resistant), "0", Resistant)) %>%
  rename(`Resistant (PPV)`=Resistant) %>%
  rename(`NWT (PPV)`=NWT) %>%
  rename(`Association with Resistance (OR, p-value)` = R_OR) %>%
  rename(`Association with NWT (OR, p-value)` = NWT_OR) %>%
  write_csv(file="tables/Table1_freq_per_det_byPheno_summary_singleDet.csv")
```




#Table S6
```{r}

#0 QRDR, 0 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==0, nonWT_binary==1))

#1 QRDR, 0 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==0, nonWT_binary==1))


#>=2 QRDR, 0 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==0, nonWT_binary==1))

#0 QRDR, 1 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==0, nonWT_binary==1))


#0 QRDR, >=2 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==0, nonWT_binary==1))


#>=1 QRDR, >=1 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==0, nonWT_binary==1))

#AAC6

#0 QRDR, 0 qnr/qep, 1 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==1, nonWT_binary==1))

#1 QRDR, 0 qnr/qep, 1 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==1, nonWT_binary==1))


#>=2 QRDR, 0 qnr/qep, 1 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==1, nonWT_binary==1))



#0 QRDR, 1 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==1, nonWT_binary==1))


#0 QRDR, >=2 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==1, nonWT_binary==1))


#>=1 QRDR, >=1 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==1, nonWT_binary==1))


# Fishers tests (Non-S)

cipro_antibiogram_noQRDRnoacquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0))
fisher.test(table(cipro_antibiogram_noQRDRnoacquired$nonWT_binary, cipro_antibiogram_noQRDRnoacquired$aac6))

cipro_antibiogram_1QRDR_noacquired<-(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0))
fisher.test(table(cipro_antibiogram_1QRDR_noacquired$nonWT_binary, cipro_antibiogram_1QRDR_noacquired$aac6))

cipro_antibiogram_2QRDR_noacquired<-(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0))
fisher.test(table(cipro_antibiogram_2QRDR_noacquired$nonWT_binary, cipro_antibiogram_2QRDR_noacquired$aac6))

cipro_antibiogram_noQRDR_1acquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1))
fisher.test(table(cipro_antibiogram_noQRDR_1acquired$nonWT_binary, cipro_antibiogram_noQRDR_1acquired$aac6))

#Infinite odds ratio
#cipro_antibiogram_noQRDR_2acquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1))
#fisher.test(table(cipro_antibiogram_noQRDR_2acquired$nonWT_binary, cipro_antibiogram_noQRDR_2acquired$aac6))

cipro_antibiogram_1QRDR_1acquired<-(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1))
fisher.test(table(cipro_antibiogram_1QRDR_1acquired$nonWT_binary, cipro_antibiogram_1QRDR_1acquired$aac6))



# Fishers tests (R)

cipro_antibiogram_noQRDRnoacquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0))
fisher.test(table(cipro_antibiogram_noQRDRnoacquired$resistant, cipro_antibiogram_noQRDRnoacquired$aac6))

cipro_antibiogram_1QRDR_noacquired<-(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0))
fisher.test(table(cipro_antibiogram_1QRDR_noacquired$resistant, cipro_antibiogram_1QRDR_noacquired$aac6))

cipro_antibiogram_2QRDR_noacquired<-(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0))
fisher.test(table(cipro_antibiogram_2QRDR_noacquired$resistant, cipro_antibiogram_2QRDR_noacquired$aac6))

cipro_antibiogram_noQRDR_1acquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1))
fisher.test(table(cipro_antibiogram_noQRDR_1acquired$resistant, cipro_antibiogram_noQRDR_1acquired$aac6))

#Infinite odds ratio
#cipro_antibiogram_noQRDR_2acquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1))
#fisher.test(table(cipro_antibiogram_noQRDR_2acquired$resistant, cipro_antibiogram_noQRDR_2acquired$aac6))

cipro_antibiogram_1QRDR_1acquired<-(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1))
fisher.test(table(cipro_antibiogram_1QRDR_1acquired$resistant, cipro_antibiogram_1QRDR_1acquired$aac6))
```

