---
title: "Mutation frequencies"
author: "Iana Amke, Kara Tsang, Kat Holt"
date: "Last compiled on `r format(Sys.time(), '%d/%m/%y')`"
output: 
  html_document: 
    df_print: paged
    fig_width: 9
    fig_height: 8
    mode: selfcontained
    highlight: pygments
    theme: 
      bootswatch: flatly
      code_font:
        google: "JetBrains Mono"
editor_options: 
  chunk_output_type: inline
---

```{=html}
<style type="text/css">
.main-container {
  max-width: 1600px !important;
  margin-left: auto;
  margin-right: auto;
}
</style>
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = 'png', echo = TRUE, message = FALSE, warning = FALSE,
                      cache = FALSE, fig.align = 'center')
knitr::opts_knit$set(root.dir = getwd())

# Data wrangling
library(tidyverse)
library(dplyr)
library(countrycode)
library(pheatmap)
library(ggpubr)
library(ggiraphExtra)
library(ggupset)
library(scales)
library(reshape2)
library(ComplexUpset)
library(RColorBrewer)
library(naniar)
library(grid)
library(gtsummary)
library(broom)
library(patchwork)
library(ggrepel)
library(magrittr)
library(epiR)

# https://krassowski.github.io/complex-upset/articles/Examples_R.html
```

# load data
``` {r}
geno <- read_delim("tables/TableS1_genotypes.tsv")

pheno <- read_delim("tables/TableS2_phenotypes_metadata.tsv")

cipro_antibiogram <- full_join(geno,pheno)

```

## long form, one row per strain/determinant combination
``` {r}
cipro_res_mech <- cipro_antibiogram %>%
  mutate(Aac6_gene=if_else(aac6 == 1, "aac(6')-Ib-cr", "-")) %>%
  pivot_longer(c(Flq_mutations,Flq_acquired, Aac6_gene), names_to = "marker.type", 
               values_to = "marker") %>%
  separate_longer_delim(cols=marker, delim=";")
```

# Summarise mutations - Table 1, Table S5

# Numbers for text - para 2
``` {r}
# total isolates
totalN <- cipro_antibiogram %>% select(strain) %>% unique() %>% nrow()
totalN

# number of isolates with each type of mutation
total_withQRDR <- cipro_res_mech %>% filter(marker.type=="Flq_mutations") %>% filter(marker!="-") %>% select(strain) %>% unique() %>% nrow()
total_withQRDR
total_withQRDR/totalN*100

total_withPMWR <- cipro_res_mech %>% filter(marker.type=="Flq_acquired") %>% filter(marker!="-") %>% select(strain) %>% unique() %>% nrow()
total_withPMWR
total_withPMWR/totalN*100

total_withAAC <- cipro_res_mech %>% filter(marker.type=="Aac6_gene") %>% filter(marker!="-") %>% select(strain) %>% unique() %>% nrow()
total_withAAC
total_withAAC/totalN*100

# freq of determinants in susceptible strains
totalS <- cipro_res_mech %>% filter(Resistance.phenotype=="S") %>% select(strain) %>% unique() %>% nrow()

S_withQRDR <- cipro_res_mech %>% filter(marker.type=="Flq_mutations") %>% filter(marker!="-") %>% filter(Resistance.phenotype=="S") %>% select(strain) %>% unique() %>% nrow()
S_withQRDR
S_withQRDR/totalS*100

S_withPMQR <- cipro_res_mech %>% filter(marker.type=="Flq_acquired") %>% filter(marker!="-") %>% filter(Resistance.phenotype=="S") %>% select(strain) %>% unique() %>% nrow()
S_withPMQR
S_withPMQR/totalS*100

S_withAac <- cipro_res_mech %>% filter(marker.type=="Aac6_gene") %>% filter(marker!="-") %>% filter(Resistance.phenotype=="S") %>% select(strain) %>% unique() %>% nrow()
S_withAac
S_withAac/totalS*100

# freq of determinants in susceptible strains
totalR <- cipro_res_mech %>% filter(Resistance.phenotype=="R") %>% select(strain) %>% unique() %>% nrow()

R_withQRDR <- cipro_res_mech %>% filter(marker.type=="Flq_mutations") %>% filter(marker!="-") %>% filter(Resistance.phenotype=="R") %>% select(strain) %>% unique() %>% nrow()
R_withQRDR
R_withQRDR/totalR*100

R_withPMQR <- cipro_res_mech %>% filter(marker.type=="Flq_acquired") %>% filter(marker!="-") %>% filter(Resistance.phenotype=="R") %>% select(strain) %>% unique() %>% nrow()
R_withPMQR
R_withPMQR/totalR*100

R_withAac <- cipro_res_mech %>% filter(marker.type=="Aac6_gene") %>% filter(marker!="-") %>% filter(Resistance.phenotype=="R") %>% select(strain) %>% unique() %>% nrow()
R_withAac
R_withAac/totalR*100
```

# function to cross-tabulate geno-pheno
``` {r}
get_phenotype_table <- function(phenotype=phenotype,  strain_list=unique(cipro_antibiogram$strain), freq_per_det=freq_per_det) {

freq_per_pheno <- cipro_antibiogram %>% 
  filter(strain %in% strain_list) %>%
  group_by(get(phenotype)) %>% 
  count() %>% rename(P1=n)

freq_per_det_byPheno <- cipro_res_mech_clean %>%
  filter(strain %in% strain_list) %>%
  group_by(marker.type, marker, get(phenotype)) %>% 
  count() %>%
  bind_rows( # add `GyrA-83`
    cipro_antibiogram %>% 
      filter(strain %in% strain_list) %>%
      filter(`GyrA-83`>0) %>%
      group_by(get(phenotype)) %>% 
      count() %>%
      mutate(marker="GyrA-83")
  ) %>%
  bind_rows( # add `GyrA-87`
    cipro_antibiogram %>% 
      filter(strain %in% strain_list) %>%
      filter(`GyrA-87`>0) %>%
      group_by(get(phenotype)) %>% 
      count() %>%
      mutate(marker="GyrA-87")
  ) %>%
  bind_rows( # add `ParC-80`
    cipro_antibiogram %>% 
      filter(strain %in% strain_list) %>%
      filter(`ParC-80`>0) %>%
      group_by(get(phenotype)) %>% 
      count() %>%
      mutate(marker="ParC-80")
  ) %>%
  bind_rows( # add `ParC-84`
    cipro_antibiogram %>% 
      filter(strain %in% strain_list) %>%
      filter(`ParC-84`>0) %>%
      group_by(get(phenotype)) %>% 
      count() %>%
      mutate(marker="ParC-84")
  ) %>%
  left_join(freq_per_det) %>% 
  left_join(freq_per_pheno) %>% 
  filter(marker != "-") %>%
  rename(G1P1=n) %>% 
  mutate(G1P0=G1-G1P1) %>%
  mutate(G0P1=P1-G1P1) %>%
  mutate(G0P0=nrow(cipro_antibiogram) - G1P1-G1P0-G0P1) %>%
  mutate(OR=(G1P1/G0P1)/(G1P0/G0P0)) %>%
  mutate(p_val=case_when(
    OR!="Inf" ~ fisher.test(x=matrix(c(G0P0, G1P0, G0P1, G1P1), ncol=2))$p.value,
    TRUE ~ NA
  )) %>%
  ungroup() %>%
  mutate(ppv = G1P1/G1) %>%
  mutate(freq_within_phenoGroup = G1P1/P1) %>%
  rename(phenotype=`get(phenotype)`) %>%
  mutate(summary=paste0("n=", G1P1, " (", round(freq_within_phenoGroup*100,2), "%)"))
  
return(table=freq_per_det_byPheno)  
}
```

# Table S5 - freq per determinant
``` {r TableS5}
# clean up determinants for analysis
cipro_res_mech_clean <- cipro_res_mech %>% 
  mutate(marker=gsub("[\\^\\*]", "", marker))

```

``` {r}
freq_per_det <- cipro_res_mech_clean %>%
  group_by(marker) %>% 
  count() %>% rename(G1=n) %>%
  bind_rows( 
    cipro_antibiogram %>% 
    select(`GyrA-83`:`ParC-84`) %>% 
    apply(2, sum) %>% 
    as_tibble(rownames="marker") %>%
    rename(G1=value)
  )

tableR <- get_phenotype_table(phenotype="Resistance.phenotype", freq_per_det=freq_per_det)

tableNTW <- get_phenotype_table(phenotype="WT_vs_nonWT", freq_per_det=freq_per_det)

# assemble summary table

# R columns
freq_per_det_byPheno_summary <- tableR %>%
  filter(phenotype=="R") %>% # get R column
  rename(Resistant=summary) %>%
  mutate(R_PPV=paste0(round(ppv*100,2),"%")) %>%
  mutate(R_OR=if_else(OR!="Inf",paste0(round(OR,2), " (p=",scientific(p_val,2), ")"), "-")) %>%
  select(marker, Resistant, R_PPV, R_OR)

# NWT columns
freq_per_det_byPheno_summary <- tableNTW %>%
  filter(phenotype=="non-wt (I/R)") %>% # add NWT column
  rename(NWT=summary) %>% 
  mutate(NWT_PPV=paste0(round(ppv*100,2),"%")) %>%
  mutate(NWT_OR=if_else(OR!="Inf",paste0(round(OR,2), " (p=",scientific(p_val,2), ")"), "-")) %>%
  select(marker, NWT, NWT_PPV, NWT_OR) %>%
  full_join(freq_per_det_byPheno_summary) 

# S columns and Gene, Total
freq_per_det_byPheno_summary <- tableR %>%
  filter(phenotype=="S") %>% # add susceptible/WT column
  rename(`Susceptible`=summary) %>% 
  full_join(freq_per_det_byPheno_summary) %>%
  filter(marker!="-") %>%
  full_join(freq_per_det, by="marker") %>%
  mutate(Total=if_else(is.na(G1.y), "0", paste0("n=", G1.y, " (", round(G1.y/nrow(cipro_antibiogram)*100,2), "%)"))) %>% 
  filter(marker!="-") %>%
  arrange(marker) %>% # sort alphabetically
  select(marker, Total, Susceptible, Resistant, R_PPV, R_OR, NWT, NWT_PPV, NWT_OR) %>%
  rename(Determinant=marker) %>%
  mutate(Susceptible=if_else(is.na(Susceptible), "0", Susceptible)) %>%
  mutate(NWT=if_else(is.na(NWT), "0", NWT)) %>%
  mutate(Resistant=if_else(is.na(Resistant), "0", Resistant)) %>%
  mutate(R_OR=if_else(is.na(R_OR),"-",R_OR)) %>%
  mutate(NWT_OR=if_else(is.na(NWT_OR),"-",NWT_OR)) %>%
  mutate(R_PPV=if_else(is.na(R_PPV),"-",R_PPV)) %>%
  mutate(NWT_PPV=if_else(is.na(NWT_PPV),"-",NWT_PPV))
```

# write out Table S5
``` {r}
# title columns for table
freq_per_det_byPheno_summary %>%
  rename(`Total (% of genomes)`=Total) %>%
  rename(`Association with Resistance (OR, p-value)`=R_OR) %>%
  rename(`Association with NWT (OR, p-value)`=NWT_OR) %>%
  rename(`Resistant (% of phenotype with determinant)`=Resistant) %>%
  rename(`Susceptible (% of phenotype with determinant)`=Susceptible) %>%
  rename(`NWT (% of phenotype with determinant)`=NWT) %>%
  rename(`PPV for Resistant`=R_PPV) %>%
  rename(`PPV for NWT`=NWT_PPV) %>%
  write_csv(file="tables/TableS5_freq_per_det_byPheno_summary.csv")
```


# Table 1
## freq stats for single QRDR mutations found in genomes without acquired genes or aac
``` {r }
strains_singleQRDR_noAcquired <- cipro_antibiogram %>%
  filter(QRDR_mutations==1 & aac6==0 & acquired_genes==0) %>%
  pull(strain)

strains_01QRDR_noAcquired <- cipro_antibiogram %>%
  filter(QRDR_mutations<=1 & aac6==0 & acquired_genes==0) %>%
  pull(strain)

freq_per_det_01QRDR_noAcquired <- cipro_res_mech_clean %>% 
  filter(strain %in% strains_01QRDR_noAcquired) %>%
  select(strain, marker) %>%
  group_by(marker) %>% 
  count() %>% 
  rename(G1=n) %>% 
  filter(marker!="-") %>%
  bind_rows( 
    cipro_antibiogram %>% 
    filter(strain %in% strains_01QRDR_noAcquired) %>%
    select(`GyrA-83`:`ParC-84`) %>% 
    apply(2, sum) %>% 
    as_tibble(rownames="marker") %>%
    rename(G1=value)
  )

tableR_soloQRDR <- get_phenotype_table(phenotype="Resistance.phenotype", strain_list=strains_01QRDR_noAcquired, freq_per_det=freq_per_det_01QRDR_noAcquired)

tableNTW_soloQRDR <- get_phenotype_table(phenotype="WT_vs_nonWT", strain_list=strains_01QRDR_noAcquired, freq_per_det=freq_per_det_01QRDR_noAcquired)

```

## freq stats for acquired genes found in genomes without QRDR or aac
``` {r}
strains_single_acquired_gene <- cipro_antibiogram %>%
  filter(QRDR_mutations==0 & aac6==0 & acquired_genes==1) %>%
  pull(strain)

strains_noQRDR_noAac <- cipro_antibiogram %>%
  filter(QRDR_mutations==0 & aac6==0) %>%
  pull(strain)

freq_per_det_noQRDR_noAac <- cipro_res_mech_clean %>% 
  filter(strain %in% strains_noQRDR_noAac) %>%
  select(strain, marker) %>%
  group_by(marker) %>% 
  count() %>% 
  rename(G1=n) %>% 
  filter(marker!="-")

tableR_soloPMQR <- get_phenotype_table(phenotype="Resistance.phenotype", strain_list=strains_noQRDR_noAac, freq_per_det=freq_per_det_noQRDR_noAac)

tableNTW_soloPMQR <- get_phenotype_table(phenotype="WT_vs_nonWT", strain_list=strains_noQRDR_noAac, freq_per_det=freq_per_det_noQRDR_noAac)

```

#### aac6 (in absence of QRDR/qnr/qep, compare to denominator of all lacking QRDR/qnr/qep)
``` {r}

strains_noQRDRqnr <- cipro_antibiogram %>%
  filter(QRDR_mutations==0 & acquired_genes==0) %>%
  pull(strain)

freq_per_det_noQRDRqnr <- cipro_res_mech_clean %>% 
  filter(strain %in% strains_noQRDRqnr) %>%
  select(strain, marker) %>%
  group_by(marker) %>% 
  count() %>% 
  rename(G1=n) %>% 
  filter(marker!="-")

tableR_soloAac6 <- get_phenotype_table(phenotype="Resistance.phenotype", strain_list=strains_noQRDRqnr, freq_per_det=freq_per_det_noQRDRqnr)

tableNTW_soloAac6 <- get_phenotype_table(phenotype="WT_vs_nonWT", strain_list=strains_noQRDRqnr, freq_per_det=freq_per_det_noQRDRqnr)
```

# assemble summary table
``` {r}

freq_per_det_byPheno_summary_solo <- tableR_soloQRDR %>%
  bind_rows(tableR_soloPMQR) %>%
  bind_rows(tableR_soloAac6) %>%
  filter(phenotype=="R") %>% # add R columns
  mutate(Resistant=paste0(G1P1," (",round(ppv*100,2),"%)")) %>%
  mutate(R_OR=if_else(OR!="Inf",paste0(round(OR,2), " (p=",scientific(p_val,2), ")"), "-")) %>%
  select(marker, Resistant, R_OR, G1)

freq_per_det_byPheno_summary_solo <- tableNTW_soloQRDR %>%
  bind_rows(tableNTW_soloPMQR) %>%
  bind_rows(tableNTW_soloAac6) %>%
  filter(phenotype=="non-wt (I/R)") %>% # add NWT column
  mutate(NWT=paste0(G1P1," (",round(ppv*100,2),"%)")) %>%
  mutate(NWT_OR=if_else(OR!="Inf",paste0(round(OR,2), " (p=",scientific(p_val,2), ")"), "-")) %>%
  select(marker, NWT, NWT_OR, G1) %>%
  full_join(freq_per_det_byPheno_summary_solo)

freq_per_det_byPheno_summary_solo <- tableR_soloQRDR %>%
  bind_rows(tableR_soloPMQR) %>%
  bind_rows(tableR_soloAac6) %>%
  filter(phenotype=="S") %>% # add S/NWT column
  rename(Susceptible=G1P1) %>%
  full_join(freq_per_det_byPheno_summary_solo) %>%
  filter(marker!="-") %>%
  rename(Total=G1) %>% 
  arrange(marker) %>% # sort by frequency of determinant
  rename(`Solo determinant`=marker) %>%
  select(`Solo determinant`, Total, Susceptible, Resistant, R_OR, NWT, NWT_OR)

```


## write out Table 1 - solo determinants and association with phenotype
``` {r}

freq_per_det_byPheno_summary_solo %>%
  mutate(Susceptible=if_else(is.na(Susceptible), 0, Susceptible)) %>%
  mutate(NWT=if_else(is.na(NWT), "0", NWT)) %>%
  mutate(Resistant=if_else(is.na(Resistant), "0", Resistant)) %>%
  mutate(R_OR=if_else(is.na(R_OR), "-", R_OR)) %>%
  mutate(NWT_OR=if_else(is.na(NWT_OR), "-", NWT_OR)) %>%
  rename(`Resistant (PPV)`=Resistant) %>%
  rename(`NWT (PPV)`=NWT) %>%
  rename(`Association with Resistance (OR, p-value)` = R_OR) %>%
  rename(`Association with NWT (OR, p-value)` = NWT_OR) %>%
  write_csv(file="tables/Table1_freq_per_det_byPheno_summary_singleDet.csv")
```


## freq stats for single QRDR mutations found in genomes without acquired genes but with aac
``` {r Table1}
strains_singleQRDR_noPMQR_aacPos <- cipro_antibiogram %>%
  filter(QRDR_mutations==1 & aac6==1 & acquired_genes==0) %>%
  pull(strain)

strains_01QRDR_noPMQR_aacPos <- cipro_antibiogram %>%
  filter(QRDR_mutations<=1 & aac6==1 & acquired_genes==0) %>%
  pull(strain)

freq_per_det_01QRDR_noPMQR_aacPos <- cipro_res_mech_clean %>% 
  filter(strain %in% strains_01QRDR_noPMQR_aacPos) %>%
  select(strain, marker) %>%
  group_by(marker) %>% 
  count() %>% 
  rename(G1=n) %>% 
  filter(marker!="-") %>%
  bind_rows( 
    cipro_antibiogram %>% 
    filter(strain %in% strains_01QRDR_noPMQR_aacPos) %>%
    select(`GyrA-83`:`ParC-84`) %>% 
    apply(2, sum) %>% 
    as_tibble(rownames="marker") %>%
    rename(G1=value)
  )

tableR_soloQRDR_aac6 <- get_phenotype_table(phenotype="Resistance.phenotype", strain_list=strains_01QRDR_noPMQR_aacPos, freq_per_det=freq_per_det_01QRDR_noPMQR_aacPos)

tableNTW_soloQRDR_aac6 <- get_phenotype_table(phenotype="WT_vs_nonWT", strain_list=strains_01QRDR_noPMQR_aacPos, freq_per_det=freq_per_det_01QRDR_noPMQR_aacPos)

```

## freq stats for acquired genes found in genomes without QRDR but with aac6
``` {r}
strains_noQRDR_Aac <- cipro_antibiogram %>%
  filter(QRDR_mutations==0 & aac6==1) %>%
  pull(strain)

freq_per_det_noQRDR_Aac <- cipro_res_mech_clean %>% 
  filter(strain %in% strains_noQRDR_Aac) %>%
  select(strain, marker) %>%
  group_by(marker) %>% 
  count() %>% 
  rename(G1=n) %>% 
  filter(marker!="-")

tableR_soloPMQR_aac6 <- get_phenotype_table(phenotype="Resistance.phenotype", strain_list=strains_noQRDR_Aac, freq_per_det=freq_per_det_noQRDR_Aac)

tableNTW_soloPMQR_aac6 <- get_phenotype_table(phenotype="WT_vs_nonWT", strain_list=strains_noQRDR_Aac, freq_per_det=freq_per_det_noQRDR_Aac)

```

# assemble summary table - QRDR/PMQR with aac6
``` {r}

freq_per_det_byPheno_summary_withAcc6 <- tableR_soloQRDR_aac6 %>%
  bind_rows(tableR_soloPMQR_aac6) %>%
  filter(phenotype=="R") %>% # add R columns
  mutate(Resistant=paste0(G1P1," (",round(ppv*100,2),"%)")) %>%
  mutate(R_OR=if_else(OR!="Inf",paste0(round(OR,2), " (p=",scientific(p_val,2), ")"), "-")) %>%
  select(marker, Resistant, R_OR, G1)

freq_per_det_byPheno_summary_withAcc6 <- tableNTW_soloQRDR_aac6 %>%
  bind_rows(tableNTW_soloPMQR_aac6) %>%
  filter(phenotype=="non-wt (I/R)") %>% # add NWT column
  mutate(NWT=paste0(G1P1," (",round(ppv*100,2),"%)")) %>%
  mutate(NWT_OR=if_else(OR!="Inf",paste0(round(OR,2), " (p=",scientific(p_val,2), ")"), "-")) %>%
  select(marker, NWT, NWT_OR, G1) %>%
  full_join(freq_per_det_byPheno_summary_withAcc6)

freq_per_det_byPheno_summary_withAcc6 <- tableR_soloQRDR_aac6 %>%
  bind_rows(tableR_soloPMQR_aac6) %>%
  filter(phenotype=="S") %>% # add S/NWT column
  rename(Susceptible=G1P1) %>%
  full_join(freq_per_det_byPheno_summary_withAcc6) %>%
  filter(marker!="-") %>%
  filter(marker!="aac(6')-Ib-cr") %>%
  rename(Total=G1) %>% 
  arrange(marker) %>% # sort by frequency of determinant
  rename(`Solo determinant`=marker) %>%
  select(`Solo determinant`, Total, Susceptible, Resistant, R_OR, NWT, NWT_OR)

```


``` {r}
freq_per_det_byPheno_summary_solo %>% 
  left_join(freq_per_det_byPheno_summary_withAcc6, by="Solo determinant") %>%
  filter(`Solo determinant`!="aac(6')-Ib-cr") %>%
  mutate(Resistant.x=if_else(is.na(Resistant.x), "0", Resistant.x)) %>%
  mutate(Resistant.y=if_else(is.na(Resistant.y), "0", Resistant.y)) %>%
  mutate(NWT.x=if_else(is.na(NWT.x), "0", NWT.x)) %>%
  mutate(NWT.y=if_else(is.na(NWT.y), "0", NWT.y)) %>%
  rename(`Solo Total`=Total.x) %>%
  rename(`Solo S` = Susceptible.x) %>%
  rename(`Solo R (PPV)`=Resistant.x) %>%
  rename(`Solo R (OR)`=R_OR.x) %>%
  rename(`Solo NWT (PPV)`=NWT.x) %>%
  rename(`Solo NWT (OR)`=NWT_OR.x) %>%
  rename(`Solo+Aac6 Total`=Total.y) %>%
  rename(`Solo+Aac6 S` = Susceptible.y) %>%
  rename(`Solo+Aac6 R (PPV)`=Resistant.y) %>%
  rename(`Solo+Aac6 R (OR)`=R_OR.y) %>%
  rename(`Solo+Aac6 NWT (PPV)`=NWT.y) %>%
  rename(`Solo+Aac6 NWT (OR)`=NWT_OR.y) %>%
  rename(`Determinant`=`Solo determinant`) %>%
  write_csv(file="tables/solo_stats_withAndWithoutAAC6.csv")
```



#Table S6
```{r}

#0 QRDR, 0 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==0, nonWT_binary==1))

#1 QRDR, 0 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==0, nonWT_binary==1))


#>=2 QRDR, 0 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==0, nonWT_binary==1))

#0 QRDR, 1 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==0, nonWT_binary==1))


#0 QRDR, >=2 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==0, nonWT_binary==1))


#>=1 QRDR, >=1 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==0))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==0, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==0, nonWT_binary==1))

#AAC6

#0 QRDR, 0 qnr/qep, 1 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0, aac6==1, nonWT_binary==1))

#1 QRDR, 0 qnr/qep, 1 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0, aac6==1, nonWT_binary==1))


#>=2 QRDR, 0 qnr/qep, 1 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0, aac6==1, nonWT_binary==1))



#0 QRDR, 1 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1, aac6==1, nonWT_binary==1))


#0 QRDR, >=2 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1, aac6==1, nonWT_binary==1))


#>=1 QRDR, >=1 qnr/qep, 0 aac6
  #total
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==1))
  #R
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==1, resistant==1))
  #R+I
nrow(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1, aac6==1, nonWT_binary==1))


# Fishers tests (Non-S)

cipro_antibiogram_noQRDRnoacquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0))
fisher.test(table(cipro_antibiogram_noQRDRnoacquired$nonWT_binary, cipro_antibiogram_noQRDRnoacquired$aac6))

cipro_antibiogram_1QRDR_noacquired<-(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0))
fisher.test(table(cipro_antibiogram_1QRDR_noacquired$nonWT_binary, cipro_antibiogram_1QRDR_noacquired$aac6))

cipro_antibiogram_2QRDR_noacquired<-(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0))
fisher.test(table(cipro_antibiogram_2QRDR_noacquired$nonWT_binary, cipro_antibiogram_2QRDR_noacquired$aac6))

cipro_antibiogram_noQRDR_1acquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1))
fisher.test(table(cipro_antibiogram_noQRDR_1acquired$nonWT_binary, cipro_antibiogram_noQRDR_1acquired$aac6))

#Infinite odds ratio
#cipro_antibiogram_noQRDR_2acquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1))
#fisher.test(table(cipro_antibiogram_noQRDR_2acquired$nonWT_binary, cipro_antibiogram_noQRDR_2acquired$aac6))

cipro_antibiogram_1QRDR_1acquired<-(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1))
fisher.test(table(cipro_antibiogram_1QRDR_1acquired$nonWT_binary, cipro_antibiogram_1QRDR_1acquired$aac6))



# Fishers tests (R)

cipro_antibiogram_noQRDRnoacquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==0))
fisher.test(table(cipro_antibiogram_noQRDRnoacquired$resistant, cipro_antibiogram_noQRDRnoacquired$aac6))

cipro_antibiogram_1QRDR_noacquired<-(cipro_antibiogram %>% filter(QRDR_mutations==1, acquired_genes==0))
fisher.test(table(cipro_antibiogram_1QRDR_noacquired$resistant, cipro_antibiogram_1QRDR_noacquired$aac6))

cipro_antibiogram_2QRDR_noacquired<-(cipro_antibiogram %>% filter(QRDR_mutations>1, acquired_genes==0))
fisher.test(table(cipro_antibiogram_2QRDR_noacquired$resistant, cipro_antibiogram_2QRDR_noacquired$aac6))

cipro_antibiogram_noQRDR_1acquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes==1))
fisher.test(table(cipro_antibiogram_noQRDR_1acquired$resistant, cipro_antibiogram_noQRDR_1acquired$aac6))

#Infinite odds ratio
#cipro_antibiogram_noQRDR_2acquired<-(cipro_antibiogram %>% filter(QRDR_mutations==0, acquired_genes>1))
#fisher.test(table(cipro_antibiogram_noQRDR_2acquired$resistant, cipro_antibiogram_noQRDR_2acquired$aac6))

cipro_antibiogram_1QRDR_1acquired<-(cipro_antibiogram %>% filter(QRDR_mutations>=1, acquired_genes>=1))
fisher.test(table(cipro_antibiogram_1QRDR_1acquired$resistant, cipro_antibiogram_1QRDR_1acquired$aac6))
```

