---
title: "Ciprofloxacin resistance in Klebsiella pneumoniae"
author: "Kara Tsang, Kat Holt"
date: "Last compiled on `r format(Sys.time(), '%d/%m/%y')`"
output: 
  html_document: 
    df_print: paged
    fig_width: 9
    fig_height: 8
    mode: selfcontained
    highlight: pygments
    theme: 
      bootswatch: flatly
      code_font:
        google: "JetBrains Mono"
editor_options: 
  chunk_output_type: inline
---

```{=html}
<style type="text/css">
.main-container {
  max-width: 1600px !important;
  margin-left: auto;
  margin-right: auto;
}
</style>
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = 'png', echo = TRUE, message = FALSE, warning = FALSE,
                      cache = FALSE, fig.align = 'center')
knitr::opts_knit$set(root.dir = getwd())

# Data wrangling
library(tidyverse)
library(dplyr)
library(reshape2)
library(ComplexUpset)
library(patchwork)
library(scales)
```

# load data
``` {r}
geno <- read_delim("tables/TableS2_genotypes.tsv")
pheno <- read_delim("tables/TableS3_phenotypes_metadata.tsv")
cipro_antibiogram <- full_join(geno,pheno) %>%
  mutate(SRnwt = if_else(Resistance.phenotype=="I", "NWT", Resistance.phenotype)) %>%
  mutate(SRnwt = factor(SRnwt, levels=c("S","NWT","R")))

```

## set colours
``` {r}
wt_colours <- c(`non-wt (I/R)`="IndianRed", `wt (S)`= "LightBlue")
res_colours <- c("I"="grey", "R"="IndianRed", "S"="LightBlue", "NWT"="grey")
```

## long form, one row per strain/determinant combination
``` {r}
cipro_res_mech <- cipro_antibiogram %>%
  mutate(Aac6_gene=if_else(aac6 == 1, "aac(6`)-Ib-cr", "-")) %>%
  pivot_longer(c(Flq_mutations,Flq_acquired, Aac6_gene), names_to = "marker.type", 
               values_to = "marker") %>%
  separate_longer_delim(cols=marker, delim=";")
```

# Numbers reported in text - para 3
``` {r}
# effect of ParC when added to GyrA-83 background (absence of any genes or GyrA-87)

# MIC data
QRDR_MIC_GyrA83background_noGenes_dat <- cipro_antibiogram %>%
    filter(aac6==0 & acquired_genes==0 & `GyrA-83`==1 & `GyrA-87`==0) %>% 
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, ParC_mutations, `ParC-80`, `ParC-84`)

wilcox.test(log2(as.numeric(Measurement)) ~ ParC_mutations, data=QRDR_MIC_GyrA83background_noGenes_dat)
summary(lm(log2(as.numeric(Measurement)) ~ ParC_mutations, data=QRDR_MIC_GyrA83background_noGenes_dat))
summary(as.numeric(QRDR_MIC_GyrA83background_noGenes_dat$Measurement)[QRDR_MIC_GyrA83background_noGenes_dat$`ParC_mutations`==1])
summary(as.numeric(QRDR_MIC_GyrA83background_noGenes_dat$Measurement)[QRDR_MIC_GyrA83background_noGenes_dat$`ParC_mutations`==0])

# disk diffusion
QRDR_DD_GyrA83background_noGenes_dat <- cipro_antibiogram %>%
  filter(aac6==0 & acquired_genes==0 & `GyrA-83`==1 & `GyrA-87`==0) %>% 
  filter(Laboratory.Typing.Method=="Disk diffusion") %>%
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, ParC_mutations, `ParC-80`, `ParC-84`)

wilcox.test(as.numeric(Measurement) ~ ParC_mutations, data=QRDR_DD_GyrA83background_noGenes_dat)
summary(lm(as.numeric(Measurement) ~ ParC_mutations, data=QRDR_DD_GyrA83background_noGenes_dat))
summary(as.numeric(QRDR_DD_GyrA83background_noGenes_dat$Measurement)[QRDR_DD_GyrA83background_noGenes_dat$ParC_mutations==1])
summary(as.numeric(QRDR_DD_GyrA83background_noGenes_dat$Measurement)[QRDR_DD_GyrA83background_noGenes_dat$ParC_mutations==0])

# test categorial phenotypes

# all genomes with no acquired genes (MIC/DD)
QRDR_noGenes_dat <- cipro_antibiogram %>%
    filter(aac6==0 & acquired_genes==0) %>% 
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, QRDR_mutations) %>%
  mutate(qrdr_bin=if_else(QRDR_mutations==0, 0, 1))

# nonWT vs presence/absence
fisher.test(table(QRDR_noGenes_dat$nonWT_binary, QRDR_noGenes_dat$qrdr_bin))

# resistance vs presence/absence
fisher.test(table(QRDR_noGenes_dat$resistant, QRDR_noGenes_dat$qrdr_bin))
```


# Numbers reported in text - para 4

## effect of presence/absence of any QRDR, in absence of any acquired genes
``` {r}

# total strains without PMQR or aac6
nrow(QRDR_noGenes_dat)

# presence of QRDR vs MIC
QRDR_MIC_noGenes_dat <- cipro_antibiogram %>%
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
    filter(aac6==0 & acquired_genes==0) %>% 
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, QRDR_mutations, `GyrA-83`, `ParC-80`) %>%
  mutate(qrdr_bin=if_else(QRDR_mutations==0, 0, 1))

wilcox.test(log2(as.numeric(Measurement)) ~ qrdr_bin, data=QRDR_MIC_noGenes_dat)
summary(lm(log2(as.numeric(Measurement)) ~ qrdr_bin, data=QRDR_MIC_noGenes_dat))
summary(as.numeric(QRDR_MIC_noGenes_dat$Measurement)[QRDR_MIC_noGenes_dat$qrdr_bin==1])
summary(as.numeric(QRDR_MIC_noGenes_dat$Measurement)[QRDR_MIC_noGenes_dat$qrdr_bin==0])

# presence of QRDR vs DD zone
QRDR_DD_noGenes_dat <- cipro_antibiogram %>%
    filter(aac6==0 & acquired_genes==0) %>% 
    filter(Laboratory.Typing.Method=="Disk diffusion") %>%
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, QRDR_mutations) %>%
  mutate(qrdr_bin=if_else(QRDR_mutations==0, 0, 1))

wilcox.test(as.numeric(Measurement) ~ qrdr_bin, data=QRDR_DD_noGenes_dat)
summary(lm(as.numeric(Measurement) ~ qrdr_bin, data=QRDR_DD_noGenes_dat))
summary(as.numeric(QRDR_DD_noGenes_dat$Measurement)[QRDR_DD_noGenes_dat$qrdr_bin==1])
summary(as.numeric(QRDR_DD_noGenes_dat$Measurement)[QRDR_DD_noGenes_dat$qrdr_bin==0])

# presence of QRDR vs nonWT
fisher.test(table(QRDR_noGenes_dat$nonWT_binary, QRDR_noGenes_dat$qrdr_bin))

# presence of QRDR vs R
fisher.test(table(QRDR_noGenes_dat$resistant, QRDR_noGenes_dat$qrdr_bin))

```

## effect of number of QRDR mutations, in absence of any acquired genes
``` {r}

# MIC vs QRDR count
QRDR_MIC_noGenes_dat <- QRDR_MIC_noGenes_dat %>%
  mutate(QRDR_0_1_2 = if_else(QRDR_mutations>2, 2, QRDR_mutations)) %>%
  mutate(QRDR_0_1_2_3 = if_else(QRDR_mutations>3, 3, QRDR_mutations)) %>%
  mutate(QRDR_1_2 = if_else(QRDR_0_1_2==0, NA, QRDR_0_1_2)) %>%
  mutate(QRDR_2_3 = if_else(QRDR_0_1_2<2, NA, QRDR_0_1_2_3))

# median MICs, grouped by QRDR count
summary(as.numeric(QRDR_MIC_noGenes_dat$Measurement)[QRDR_MIC_noGenes_dat$QRDR_mutations==0])
summary(as.numeric(QRDR_MIC_noGenes_dat$Measurement)[QRDR_MIC_noGenes_dat$QRDR_mutations==1])
summary(as.numeric(QRDR_MIC_noGenes_dat$Measurement)[QRDR_MIC_noGenes_dat$QRDR_mutations>2])

# test for difference in MIC with QRDR count
summary(lm(log2(as.numeric(Measurement)) ~ factor(QRDR_0_1_2), data=QRDR_MIC_noGenes_dat))
summary(lm(log2(as.numeric(Measurement)) ~ QRDR_mutations, data=QRDR_MIC_noGenes_dat))


# DD vs QRDR count
QRDR_DD_noGenes_dat <- QRDR_DD_noGenes_dat %>%
  mutate(QRDR_0_1_2 = if_else(QRDR_mutations>2, 2, QRDR_mutations)) %>%
  mutate(QRDR_0_1_2_3 = if_else(QRDR_mutations>3, 3, QRDR_mutations)) %>%
  mutate(QRDR_1_2 = if_else(QRDR_0_1_2==0, NA, QRDR_0_1_2)) %>%
  mutate(QRDR_2_3 = if_else(QRDR_0_1_2<2, NA, QRDR_0_1_2_3))

# median DD zones, grouped by QRDR count
summary(as.numeric(QRDR_DD_noGenes_dat$Measurement)[QRDR_DD_noGenes_dat$QRDR_mutations==0])
summary(as.numeric(QRDR_DD_noGenes_dat$Measurement)[QRDR_DD_noGenes_dat$QRDR_mutations==1])
summary(as.numeric(QRDR_DD_noGenes_dat$Measurement)[QRDR_DD_noGenes_dat$QRDR_mutations>2])

# test for difference in DD zone with QRDR count
summary(lm(log2(as.numeric(Measurement)) ~ factor(QRDR_0_1_2), data=QRDR_DD_noGenes_dat))
summary(lm(log2(as.numeric(Measurement)) ~ QRDR_mutations, data=QRDR_DD_noGenes_dat))

```


## Fig1 - number of QRDR vs MIC, facet aac6 yes/no (no other genes)
``` {r}

#MIC distribution for # QRDR, in absence of genes
QRDR_MIC_noGenes <- cipro_antibiogram %>%
    filter(acquired_genes==0) %>% 
    filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>% 
    mutate(aac6_label=if_else(aac6==0, "aac(6`)-Ib-cr absent", "aac(6`)-Ib-cr present")) %>%
    mutate(QRDR_mutations=if_else(QRDR_mutations==4, 3, QRDR_mutations)) %>% # single isolate with 4 QRDR
    ggplot(aes(x=factor(QRDR_mutations), y=as.numeric(Measurement))) + 
    geom_violin() + 
    geom_count(aes(colour = SRnwt)) + 
    geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
    geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
    facet_wrap(vars(aac6_label)) +
    scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:9), labels = function(x) round(as.numeric(x), digits = 3)) +
    scale_color_manual(values = res_colours) + 
    theme_light() + 
    labs(y="MIC (mg/L)", x="", col="Phenotype", 
         title="a) No. QRDR mutations vs phenotype",
         subtitle="(in absence of PMQR genes)")

QRDR_MIC_noGenes
```

``` {r}

QRDR_pheno_noGenes <- cipro_antibiogram %>%
    filter(acquired_genes==0) %>% 
    filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
    mutate(aac6_label=if_else(aac6==0, "aac(6`)-Ib-cr absent", "aac(6`)-Ib-cr present")) %>%
    mutate(QRDR_mutations=if_else(QRDR_mutations==4, 3, QRDR_mutations)) %>% # single isolate with 4 QRDR
    ggplot(aes(x=factor(QRDR_mutations), fill=SRnwt)) + 
    geom_bar(stat='count', position='fill') + 
    facet_wrap(vars(aac6_label)) +
    scale_fill_manual(values = res_colours) + 
    scale_y_continuous(labels=scales::percent_format()) +
    geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), size=2) +
    theme_light() + 
    labs(y="% Phenotype", x="Number of QRDR mutations") + 
    theme(legend.position="none", strip.background = element_blank(), strip.text = element_blank())
```

# Fig S6 - number of QRDR vs DD, facet aac6 yes/no (no other genes)
``` {r}

# DD distribution for # QRDR, in absence of genes
QRDR_DD_noGenes <- cipro_antibiogram %>%
    filter(acquired_genes==0) %>% 
    filter(Laboratory.Typing.Method=="Disk diffusion") %>% 
    mutate(aac6_label=if_else(aac6==0, "aac(6`)-Ib-cr absent", "aac(6`)-Ib-cr present")) %>%
    mutate(QRDR_mutations=if_else(QRDR_mutations==4, 3, QRDR_mutations)) %>% # single isolate with 4 QRDR
    ggplot(aes(x=factor(QRDR_mutations), y=as.numeric(Measurement))) + 
    geom_violin() + 
    geom_count(aes(colour = SRnwt)) + 
    geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
    geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
    facet_wrap(vars(aac6_label)) +
    scale_colour_manual(values = res_colours) + 
    theme_light() + 
    labs(y="Disk diffusion\n zone diameter (mm)", x="", col="Phenotype",
         title="No. QRDR mutations vs phenotype",
         subtitle="(in absence of PMQR genes)")

QRDR_DDpheno_noGenes <- cipro_antibiogram %>%
    filter(acquired_genes==0) %>% 
    filter(Laboratory.Typing.Method=="Disk diffusion") %>%
    mutate(aac6_label=if_else(aac6==0, "aac(6`)-Ib-cr absent", "aac(6`)-Ib-cr present")) %>%
    mutate(QRDR_mutations=if_else(QRDR_mutations==4, 3, QRDR_mutations)) %>% # single isolate with 4 QRDR
    ggplot(aes(x=factor(QRDR_mutations), fill=SRnwt)) + 
    geom_bar(stat='count', position='fill') + 
    facet_wrap(vars(aac6_label)) +
    scale_fill_manual(values = res_colours) +  
    scale_y_continuous(labels=scales::percent_format()) +
    geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), size=2) +
    theme_light() + 
    labs(y="% Phenotype", x="Number of QRDR mutations") + 
    theme(legend.position="none", strip.background = element_blank(), strip.text = element_blank())

```

``` {r}
QRDR_DD_noGenes / QRDR_DDpheno_noGenes + plot_layout(heights=c(2,1))

ggsave("figs/FigS6_numQRDR_DD_pheno.pdf", width=6, height=4)
ggsave("figs/FigS6_numQRDR_DD_pheno.png", width=6, height=4)
```

# Fig S4 panel a - upset plot of QRDR positions, absence of aac6/acquired genes, with MIC data
``` {r}
upset_QRDR_noAac6_MIC <- cipro_antibiogram %>% 
  filter(aac6==0 & acquired_genes==0) %>%
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
  upset(
      intersect=c("ParC-84","ParC-80","GyrA-87","GyrA-83","wt_QRDR"), # counts are converted to binary
        sort_sets=FALSE,
      intersections=list(
        'wt_QRDR',
        "GyrA-83",
        "GyrA-87",
        c("GyrA-83","ParC-80"),
        c("GyrA-83","ParC-84"),
        c("GyrA-83","GyrA-87","ParC-80"),
        c("GyrA-83","GyrA-87","ParC-84")),  
        sort_intersections=FALSE,
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45),axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing WT/nonWT
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
                 scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_colour_manual(values = res_colours) + 
              scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:9), labels = function(x) round(as.numeric(x), digits = 3), limits=c(2^-5,2^9)) +
              theme(legend.position = "none", axis.title=element_text(size=8))  +
              ggtitle("aac(6`)-Ib-cr absent")
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

upset_QRDR_noAac6_MIC
```


# Fig S4 panel b - upset plot of QRDR positions, presence of aac6 but no other acquired genes, with MIC data
``` {r}
upset_QRDR_aac6_MIC <- cipro_antibiogram %>% 
  filter(aac6==1 & acquired_genes==0) %>%
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
  upset(
      intersect=c("ParC-84","ParC-80","GyrA-87","GyrA-83","wt_QRDR"), # counts are converted to binary
        sort_sets=FALSE,
      intersections=list(
        'wt_QRDR',
        "GyrA-83",
        "GyrA-87",
        c("GyrA-83","ParC-80"),
        c("GyrA-83","ParC-84"),
        c("GyrA-83","GyrA-87","ParC-80"),
        c("GyrA-83","GyrA-87","ParC-84")),  
        sort_intersections=FALSE,
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45),axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing WT/nonWT
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
              scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_colour_manual(values = res_colours) + 
              scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:9), labels = function(x) round(as.numeric(x), digits = 3), limits=c(2^-5,2^9)) +
              labs(colour = "Phenotype") +
              theme(axis.title=element_text(size=8)) +
              ggtitle("aac(6`)-Ib-cr present")
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas
upset_QRDR_aac6_MIC
```

# Fig S5 panel a - QRDR positions, absence of aac6/acquired genes, with DD data
``` {r}
upset_QRDR_noAac6_DD <- cipro_antibiogram %>% 
  filter(aac6==0 & acquired_genes==0) %>%
  filter(Laboratory.Typing.Method == "Disk diffusion") %>%
  upset(
      intersect=c("ParC-84","ParC-80","GyrA-87","GyrA-83","wt_QRDR"), # counts are converted to binary
        sort_sets=FALSE,
      intersections=list(
        'wt_QRDR',
        "GyrA-83",
        "GyrA-87",
        c("GyrA-83","ParC-80"),
        c("GyrA-83","GyrA-87","ParC-80"),
        c("GyrA-83","GyrA-87","ParC-84"),
        c("GyrA-83","GyrA-87","ParC-80","ParC-84")),  
        sort_intersections=FALSE,
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45), axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
             scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            theme(legend.position = "none", axis.title=element_text(size=8)),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with disk diffusion data
        'Disk diffusion zone diameter (mm)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
              geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_colour_manual(values = res_colours) + 
              guides(size = "none") +
              labs(linetype = "CLSI/EUCAST", colour = "Phenotype") +
              ggtitle("Disk diffusion data") +
              theme(axis.title=element_text(size=8), legend.position="none") + 
              ggtitle("aac(6`)-Ib-cr absent") + ylim(5,50)
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

upset_QRDR_noAac6_DD

```

# Fig S5 panel b - QRDR positions, presence of aac6 but no other acquired genes, with DD data
``` {r}
upset_QRDR_aac6_DD <- cipro_antibiogram %>% 
  filter(aac6==1 & acquired_genes==0) %>%
  filter(Laboratory.Typing.Method == "Disk diffusion") %>%
  upset(
      intersect=c("ParC-84","ParC-80","GyrA-87","GyrA-83","wt_QRDR"), # counts are converted to binary
        sort_sets=FALSE,
      intersections=list(
        'wt_QRDR',
        "GyrA-83",
        "GyrA-87",
        c("GyrA-83","ParC-80"),
        c("GyrA-83","GyrA-87","ParC-80"),
        c("GyrA-83","GyrA-87","ParC-84"),
        c("GyrA-83","GyrA-87","ParC-80","ParC-84")),  
        sort_intersections=FALSE,      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45), axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
               scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            theme(legend.position = "none", axis.title=element_text(size=8)),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with disk diffusion data
        'Disk diffusion zone diameter (mm)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
              geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_color_manual(values = c("I"="grey", "R"="IndianRed", "S"="LightBlue")) + 
              guides(size = "none") +
              labs(linetype = "CLSI/EUCAST", colour = "Phenotype") +
              ggtitle("aac(6`)-Ib-cr present") +
              theme(axis.title=element_text(size=8)) + ylim(5,50)
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

upset_QRDR_aac6_DD

```

## Supp Figs 4 and 5: QRDR sites vs pheno (MIC & DD, with/without aac6)
```{r}

upset_QRDR_noAac6_MIC | upset_QRDR_aac6_MIC

ggsave("figs/FigS4_QRDR_MIC.pdf", width=9, height=6)
ggsave("figs/FigS4_QRDR_MIC.png", width=9, height=6)


upset_QRDR_noAac6_DD | upset_QRDR_aac6_DD

ggsave("figs/FigS5_QRDR_DD.pdf", width=9, height=6)
ggsave("figs/FigS5_QRDR_DD.png", width=9, height=6)

```

## set up binary variables for common PMQR genes
``` {r}

# genes in at least 10 genomes
qnr_genes <- cipro_res_mech %>% 
  mutate(marker=gsub("[\\^\\*\\?]", "", marker)) %>%
  mutate(marker=gsub(".v1", "", marker)) %>%
  mutate(marker=gsub(".v2", "", marker)) %>%
  group_by(marker) %>% 
  count() %>% filter(startsWith(marker, "q")) %>% 
  filter(n>10) %>% pull(marker)

qnr_presence <- cipro_res_mech %>% 
  mutate(marker=gsub("[\\^\\*\\?]", "", marker)) %>%
  mutate(marker=gsub(".v1", "", marker)) %>%
  mutate(marker=gsub(".v2", "", marker)) %>%
  filter(marker %in% qnr_genes) %>%
  group_by(strain, marker) %>% count() %>% pivot_wider(names_from=marker, values_from=n)

cipro_antibiogram <- left_join(cipro_antibiogram, qnr_presence)

```


# Fig S7 panel a - upset plot of qnr genes, absence of aac6 and QRDR, with MIC data
``` {r}
upset_qnr_noAac6_MIC <- cipro_antibiogram %>% 
  filter(aac6==0 & QRDR_mutations==0) %>%
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
  upset(
      intersect=rev(qnr_genes), 
      sort_sets=FALSE,
        intersections=list(
        'Outside of known sets',
        "qnrA1",
        "qnrB17",
        "qnrB19",
        "qnrB4",
        "qnrB6",
        "qnrS1",
        c("qnrB4","qnrS1"),
        c("qnrB6","qnrS1"),
        c("qnrB6","qnrB19")),  
        sort_intersections=FALSE,
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45),axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
             scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
              scale_color_manual(values = res_colours)+
              scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:6), labels = function(x) round(as.numeric(x), digits = 3), limits=c(2^-5,2^6)) +
              theme(legend.position = "none", axis.title=element_text(size=8))  +
              ggtitle("aac(6`)-Ib-cr absent")
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas
upset_qnr_noAac6_MIC
```

# Fig S7 panel- upset plot of qnr genes, presence of aac6 but no QRDR, with MIC data
``` {r}
upset_qnr_aac6_MIC <- cipro_antibiogram %>% 
  filter(aac6==1 & QRDR_mutations==0) %>%
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
  upset(
      intersect=rev(qnr_genes), 
      sort_sets=FALSE,
        intersections=list(
        'Outside of known sets',
        "qnrA1",
        "qnrB17",
        "qnrB19",
        "qnrB4",
        "qnrB6",
        "qnrS1",
        c("qnrB4","qnrS1"),
        c("qnrB6","qnrS1"),
        c("qnrB6","qnrB19")),  
        sort_intersections=FALSE,
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45),axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
             scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
                scale_color_manual(values = res_colours)+
              scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:6), labels = function(x) round(as.numeric(x), digits = 3), limits=c(2^-5,2^6)) +
              labs(colour = "Phenotype") +
              theme(axis.title=element_text(size=8)) +
              ggtitle("aac(6`)-Ib-cr present")
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas
upset_qnr_aac6_MIC
```

# Fig S8 panel - qnr genes, absence of aac6 and QRDR mutations, with DD data
``` {r}
upset_qnr_noAac6_DD <- cipro_antibiogram %>% 
  filter(aac6==0 & QRDR_mutations==0) %>%
  filter(Laboratory.Typing.Method == "Disk diffusion") %>%
  upset(
      intersect=rev(qnr_genes), 
      sort_sets=FALSE,
        intersections=list(
        'Outside of known sets',
        "qnrA1",
        "qnrB17",
        "qnrB19",
        "qnrB4",
        "qnrB6",
        "qnrS1",
        c("qnrB4","qnrS1"),
        c("qnrB6","qnrS1"),
        c("qnrB6","qnrB19")),  
        sort_intersections=FALSE,
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45), axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
               scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            theme(legend.position = "none", axis.title=element_text(size=8)),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with disk diffusion data
        'Disk diffusion zone diameter (mm)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
              geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
                scale_color_manual(values = res_colours)+
              guides(size = "none") +
              labs(linetype = "CLSI/EUCAST", colour = "Phenotype") +
              ggtitle("Disk diffusion data") +
              theme(axis.title=element_text(size=8), legend.position="none") + 
              ggtitle("aac(6`)-Ib-cr absent") + ylim(5,50)
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

upset_qnr_noAac6_DD

```

# Fig S8 panel - qnr genes, presence of aac6 but no QRDR mutations, with DD data
``` {r}
upset_qnr_aac6_DD <- cipro_antibiogram %>% 
  filter(aac6==1 & QRDR_mutations==0) %>%
  filter(Laboratory.Typing.Method == "Disk diffusion") %>%
  upset(
      intersect=rev(qnr_genes), 
      sort_sets=FALSE,
        intersections=list(
        'Outside of known sets',
        "qnrA1",
        "qnrB17",
        "qnrB19",
        "qnrB4",
        "qnrB6",
        "qnrS1",
        c("qnrB4","qnrS1"),
        c("qnrB6","qnrS1"),
        c("qnrB6","qnrB19")),  
        sort_intersections=FALSE,
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45), axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
               scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            theme(legend.position = "none", axis.title=element_text(size=8)),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with disk diffusion data
        'Disk diffusion zone diameter (mm)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
              geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
                scale_color_manual(values = res_colours)+
              guides(size = "none") +
              labs(linetype = "CLSI/EUCAST", colour = "Phenotype") +
              ggtitle("aac(6`)-Ib-cr present") +
              theme(axis.title=element_text(size=8)) + ylim(5,50)
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

upset_qnr_aac6_DD

```

# Supp Figs S7 and S8: qnr genes vs pheno (MIC & DD, with/without aac6)
```{r}

upset_qnr_noAac6_MIC | upset_qnr_aac6_MIC

ggsave("figs/FigS7_qnr_MIC.pdf", width=12, height=6)
ggsave("figs/FigS7_qnr_MIC.png", width=12, height=6)


upset_qnr_noAac6_DD | upset_qnr_aac6_DD

ggsave("figs/FigS8_qnr_DD.pdf", width=12, height=6)
ggsave("figs/FigS8_qnr_DD.png", width=12, height=6)

```


# numbers for text - paragraph 5
``` {r}

# effect of qnr/qep genes, in absence of QRDR and aac6

# MIC data in absence of QRDR and aac6
qnr_MIC_nullBG_dat <- cipro_antibiogram %>%
  filter(Laboratory.Typing.Method !="Disk diffusion") %>%
  filter(aac6==0 & QRDR_mutations==0) %>% 
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, acquired_genes) %>%
  mutate(acquired_bin=if_else(acquired_genes==0, 0, 1))

# MIC vs presence/absence of qnr, in absence of QRDR and aac6
summary(as.numeric(qnr_MIC_nullBG_dat$Measurement)[qnr_MIC_nullBG_dat$acquired_bin==1])
summary(as.numeric(qnr_MIC_nullBG_dat$Measurement)[qnr_MIC_nullBG_dat$acquired_bin==0])

wilcox.test(log2(as.numeric(Measurement)) ~ acquired_bin, data=qnr_MIC_nullBG_dat)
summary(lm(log2(as.numeric(Measurement)) ~ acquired_bin, data=qnr_MIC_nullBG_dat))


# DD data in absence of QRDR and aac6
qnr_DD_noGenes_dat <- cipro_antibiogram %>%
    filter(aac6==0 & QRDR_mutations==0) %>% 
    filter(Laboratory.Typing.Method=="Disk diffusion") %>%
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, acquired_genes) %>%
  mutate(acquired_bin=if_else(acquired_genes==0, 0, 1))

# DD vs presence/absence of qnr, in absence of QRDR and aac6

summary(as.numeric(qnr_DD_noGenes_dat$Measurement)[qnr_DD_noGenes_dat$acquired_bin==1])
summary(as.numeric(qnr_DD_noGenes_dat$Measurement)[qnr_DD_noGenes_dat$acquired_bin==0])

wilcox.test(as.numeric(Measurement) ~ acquired_bin, data=qnr_DD_noGenes_dat)
summary(lm(as.numeric(Measurement) ~ acquired_bin, data=qnr_DD_noGenes_dat))

# all genomes with no acquired genes (MIC/DD)
qnr_noGenes_dat <- cipro_antibiogram %>%
    filter(aac6==0 & QRDR_mutations==0) %>% 
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, acquired_genes) %>%
  mutate(acquired_bin=if_else(acquired_genes==0, 0, 1))

dim(qnr_noGenes_dat)

# NWT vs presence/absence of qnr, in absence of QRDR and aac6
fisher.test(table(qnr_noGenes_dat$nonWT_binary, qnr_noGenes_dat$acquired_bin))

# resistance vs presence/absence of qnr, in absence of QRDR and aac6
fisher.test(table(qnr_noGenes_dat$resistant, qnr_noGenes_dat$acquired_bin))


# MIC vs qnr count
summary(lm(log2(as.numeric(Measurement)) ~ acquired_genes, data=qnr_MIC_nullBG_dat))

summary(as.numeric(qnr_MIC_nullBG_dat$Measurement)[qnr_MIC_nullBG_dat$acquired_genes==1])
summary(as.numeric(qnr_MIC_nullBG_dat$Measurement)[qnr_MIC_nullBG_dat$acquired_genes==2])
summary(as.numeric(qnr_MIC_nullBG_dat$Measurement)[qnr_MIC_nullBG_dat$acquired_genes>2])

# DD vs qnr count
summary(lm(as.numeric(Measurement) ~ acquired_genes, data=qnr_DD_noGenes_dat))
summary(as.numeric(qnr_DD_noGenes_dat$Measurement)[qnr_DD_noGenes_dat$acquired_genes==1])
summary(as.numeric(qnr_DD_noGenes_dat$Measurement)[qnr_DD_noGenes_dat$acquired_genes==2])
summary(as.numeric(qnr_DD_noGenes_dat$Measurement)[qnr_DD_noGenes_dat$acquired_genes>2])

```

# Figure 2 - number of qnr/qep vs MIC, facet aac6 yes/no (no QRDR)
``` {r}

# MIC distribution for # QRDR, in absence of genes
qnr_MIC_noGenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0) %>% 
    filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>% 
    mutate(aac6_label=if_else(aac6==0, "aac(6`)-Ib-cr absent", "aac(6`)-Ib-cr present")) %>%
    ggplot(aes(x=factor(acquired_genes), y=as.numeric(Measurement))) + 
    geom_violin() + 
    geom_count(aes(colour = SRnwt)) + 
    geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
    geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
    facet_wrap(vars(aac6_label)) +
    scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:9), labels = function(x) round(as.numeric(x), digits = 3)) +
    scale_color_manual(values = res_colours) + 
    theme_light() + 
    labs(y="MIC (mg/L)", x="", col="Phenotype", 
         title="b) No. PMQR genes vs phenotype",
         subtitle="(in absence of QRDR mutations)")


qnr_pheno_noGenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0) %>% 
    filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
    mutate(aac6_label=if_else(aac6==0, "aac(6`)-Ib-cr absent", "aac(6`)-Ib-cr present")) %>%
    ggplot(aes(x=factor(acquired_genes), fill=SRnwt)) + 
    geom_bar(stat='count', position='fill') + 
    facet_wrap(vars(aac6_label)) +
       scale_fill_manual(values = res_colours) + 
    scale_y_continuous(labels=scales::percent_format()) +
    geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), size=2) +
    theme_light() + 
    labs(y="% Phenotype", x="Number of PMQR genes") + 
    theme(legend.position="none", strip.background = element_blank(), strip.text = element_blank())

```

# Fig 2
``` {r}
QRDR_MIC_noGenes / QRDR_pheno_noGenes / qnr_MIC_noGenes / qnr_pheno_noGenes + plot_layout(heights=c(3,1, 3, 1))

ggsave("figs/Fig2_numQnr_numPMQR_MIC_pheno.pdf", width=6, height=10)
ggsave("figs/Fig2_numQnr_numPMQR_MIC_pheno.png", width=6, height=10)

```

# Fig S9 - number of qnr vs DD, facet aac6 yes/no (no QRDR)
``` {r}

# DD distribution for # qnr, in absence of genes
qnr_DD_noGenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0) %>% 
    filter(Laboratory.Typing.Method=="Disk diffusion") %>% 
    mutate(aac6_label=if_else(aac6==0, "aac(6`)-Ib-cr absent", "aac(6`)-Ib-cr present")) %>%
    ggplot(aes(x=factor(acquired_genes), y=as.numeric(Measurement))) + 
    geom_violin() + 
    geom_count(aes(colour = SRnwt)) + 
    geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
    geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
    facet_wrap(vars(aac6_label)) +
      scale_color_manual(values = res_colours) + 
    theme_light() + 
    labs(y="Disk diffusion\n zone diameter (mm)", x="", col="Phenotype",
         title="No. PMQR genes vs phenotype",
         subtitle="(in absence of QRDR mutations)")


qnr_DDpheno_noGenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0) %>% 
    filter(Laboratory.Typing.Method=="Disk diffusion") %>%
    mutate(aac6_label=if_else(aac6==0, "aac(6`)-Ib-cr absent", "aac(6`)-Ib-cr present")) %>%
    ggplot(aes(x=factor(acquired_genes), fill=SRnwt)) + 
    geom_bar(stat='count', position='fill') + 
    facet_wrap(vars(aac6_label)) +
      scale_fill_manual(values = res_colours) + 
    scale_y_continuous(labels=scales::percent_format()) +
    geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), size=2) +
    theme_light() + 
    labs(y="% Phenotype", x="Number of PMQR genes") + 
    theme(legend.position="none", strip.background = element_blank(), strip.text = element_blank())
```

``` {r}
qnr_DD_noGenes / qnr_DDpheno_noGenes + plot_layout(heights=c(2,1))

ggsave("figs/FigS9_numQnr_DD_pheno.pdf", width=6, height=4)
ggsave("figs/FigS9_numQnr_DD_pheno.png", width=6, height=4)
```


# Fig3 panel b - aac6, QRDR mutations, qnr with DD data
``` {r}
upset_aac6_simple_DD <- cipro_antibiogram %>% 
  filter(Laboratory.Typing.Method == "Disk diffusion") %>%
  upset(
      intersect=c("aac6","acquired_genes", "QRDR_mutations"),# counts are converted to binary
      sort_sets=FALSE,
      set_sizes=(
        upset_set_size()
        + theme(axis.text.x=element_text(angle=45))),
      intersections=list(
        'Outside of known sets',
        "aac6",
        "acquired_genes",
        'QRDR_mutations',
        c("aac6","acquired_genes"),
        c("aac6","QRDR_mutations"),
        c("acquired_genes","QRDR_mutations"),
        c("aac6","acquired_genes","QRDR_mutations")),
            sort_intersections=FALSE,
        labeller=ggplot2::as_labeller(c(
        'QRDR_mutations'='QRDR mutations',
        'acquired_genes'='PMQR genes', 
        'aac6'="aac(6`)-Ib-cr")),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
             scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            theme(legend.position = "none", axis.title=element_text(size=8)),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with disk diffusion data
        'Disk diffusion zone diameter (mm)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
              geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
              scale_color_manual(values = res_colours)+
              guides(size = "none") +
              labs(linetype = "CLSI/EUCAST", colour = "Phenotype") +
              ggtitle("b) Disk diffusion data") +
              ylim(5,50)
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

upset_aac6_simple_DD

```

# Fig3 panel a - aac6, QRDR mutations, qnr with MIC data
``` {r}
upset_aac6_simple_MIC <- cipro_antibiogram %>% 
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
  upset(
      intersect=c("aac6","acquired_genes", "QRDR_mutations"),# counts are converted to binary
      sort_sets=FALSE,
      set_sizes=(
        upset_set_size()
        + theme(axis.text.x=element_text(angle=45))),
      intersections=list(
        'Outside of known sets',
        "aac6",
        "acquired_genes",
        'QRDR_mutations',
        c("aac6","acquired_genes"),
        c("aac6","QRDR_mutations"),
        c("acquired_genes","QRDR_mutations"),
        c("aac6","acquired_genes","QRDR_mutations")),
        labeller=ggplot2::as_labeller(c(
        'QRDR_mutations'='QRDR mutations',
        'acquired_genes'='PMQR genes', 
        'aac6'="aac(6`)-Ib-cr")),   
      sort_intersections=FALSE,
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
             scale_fill_manual(values = res_colours) ,
            scale_y_continuous(labels=scales::percent_format()),
            theme(legend.position = "none", axis.title=element_text(size=8)),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 1), linetype = 1, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 0.25), linetype = 2, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_color_manual(values = res_colours)+
              scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:6), labels = function(x) round(as.numeric(x), digits = 3), limits=c(2^-5,2^6)) +
              labs(colour = "Phenotype") +
              theme(legend.position="none") + 
              ggtitle("a) MIC data")
        )

      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

upset_aac6_simple_MIC

```

## Fig3: aac6, QRDR mutations, qnr with MIC & DD data
```{r}

upset_aac6_simple_MIC | upset_aac6_simple_DD

ggsave("figs/Fig3_simple_comb_MIC_DD.pdf", width=12, height=6)
ggsave("figs/Fig3_simple_comb_MIC_DD.png", width=12, height=6)

```

# numbers for text - paragraph 6, effect of aac6 gene, in absence of QRDR and other acquired
``` {r}

# MIC
aac_MIC_nullBG_dat <- cipro_antibiogram %>%
  filter(Laboratory.Typing.Method !="Disk diffusion") %>%
    filter(acquired_genes==0 & QRDR_mutations==0) %>% 
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, aac6)

# MIC vs presence/absence of qnr
wilcox.test(log2(as.numeric(Measurement)) ~ aac6, data=aac_MIC_nullBG_dat)
summary(lm(log2(as.numeric(Measurement)) ~ aac6, data=aac_MIC_nullBG_dat))

summary(as.numeric(aac_MIC_nullBG_dat$Measurement)[aac_MIC_nullBG_dat$aac6==1])
summary(as.numeric(aac_MIC_nullBG_dat$Measurement)[aac_MIC_nullBG_dat$aac6==0])

aac_MIC_nullBG_dat %>% group_by(aac6) %>% summarise(S=mean(Resistance.phenotype=="S"), I=mean(Resistance.phenotype=="I"), R=mean(Resistance.phenotype=="R"))

# DD
aac_DD_nullBG_dat <- cipro_antibiogram %>%
    filter(acquired_genes==0 & QRDR_mutations==0) %>% 
    filter(Laboratory.Typing.Method=="Disk diffusion") %>%
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, aac6)

wilcox.test(as.numeric(Measurement) ~ aac6, data=aac_DD_nullBG_dat)
summary(lm(as.numeric(Measurement) ~ aac6, data=aac_DD_nullBG_dat))

summary(as.numeric(aac_DD_nullBG_dat$Measurement)[aac_DD_nullBG_dat$aac6==1])
summary(as.numeric(aac_DD_nullBG_dat$Measurement)[aac_DD_nullBG_dat$aac6==0])

aac_DD_nullBG_dat %>% group_by(aac6) %>% summarise(S=mean(Resistance.phenotype=="S"), I=mean(Resistance.phenotype=="I"), R=mean(Resistance.phenotype=="R"))


# all genomes with no acquired genes (MIC/DD)
qnr_noGenes_dat <- cipro_antibiogram %>%
    filter(acquired_genes==0 & QRDR_mutations==0) %>% 
  select(Measurement, Resistance.phenotype, resistant, nonWT_binary, aac6)

dim(qnr_noGenes_dat)

```
