---
title: "Ciprofloxacin resistance in Klebsiella pneumoniae"
author: "Kara Tsang, Kat Holt"
date: "Last compiled on `r format(Sys.time(), '%d/%m/%y')`"
output: 
  html_document: 
    df_print: paged
    fig_width: 9
    fig_height: 8
    mode: selfcontained
    highlight: pygments
    theme: 
      bootswatch: flatly
      code_font:
        google: "JetBrains Mono"
editor_options: 
  chunk_output_type: inline
---

```{=html}
<style type="text/css">
.main-container {
  max-width: 1600px !important;
  margin-left: auto;
  margin-right: auto;
}
</style>
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = 'png', echo = TRUE, message = FALSE, warning = FALSE,
                      cache = FALSE, fig.align = 'center')
knitr::opts_knit$set(root.dir = getwd())

# Data wrangling
library(tidyverse)
library(dplyr)
library(reshape2)
library(ComplexUpset)
library(patchwork)
library(scales)
```

# load data
``` {r}
geno <- read_delim("tables/TableS2_genotypes.tsv")
pheno <- read_delim("tables/TableS3_phenotypes_metadata.tsv")
oqxA_coverage <- read_delim("tables/oqxAB_operon_defects/oqxA_coverage.tsv")
oqxB_coverage <- read_delim("tables/oqxAB_operon_defects/oqxB_coverage.tsv")
oqxR_coverage <- read_delim("tables/oqxAB_operon_defects/oqxR_coverage.tsv")
norB_coverage <- read_delim("tables/oqxAB_operon_defects/norB_coverage.tsv")
rarA_coverage <- read_delim("tables/oqxAB_operon_defects/rarA_coverage.tsv")
rob_coverage <- read_delim("tables/oqxAB_operon_defects/rob_coverage.tsv")
oqxR_mutation <- read_delim("tables/oqxAB_operon_defects/oqxR_amrfinder.txt")

oqxab_IS_cipro <- read_delim("tables/oqxAB_operon_defects/oqxA_isfinder_filtered.tsv")
```

## set colours
``` {r}
wt_colours <- c(`non-wt (I/R)`="IndianRed", `wt (S)`= "LightBlue")
res_colours <- c("I"="grey", "R"="IndianRed", "S"="LightBlue", "NWT"="grey")
```

## Data formatting

``` {r norB truncation}
# norB truncation is defined as a genome with all norB copies with < 90% acid sequence coverage

norB_truncation <- norB_coverage %>% 
  mutate(norB_truncation=ifelse(norB_coverage < 90, 1, 0)) %>% 
  subset(select = -c(norB_coverage)) %>%
  # consider any genome with at least one complete norB as no norB truncation
  arrange(strain, match(norB_truncation, c(0, 1))) %>%
  distinct(strain, .keep_all = TRUE)

# norB not identified in nucleotide BLAST search (i.e., norB deletion) considered to be truncation
geno_norB_truncation <- full_join(geno,norB_truncation) %>%
   mutate(norB_truncation=replace_na(norB_truncation, 1))

```


``` {r rarA truncation}

# rarA truncation is defined as a genome with all rarA copies with < 90% acid sequence coverage

rarA_truncation <- rarA_coverage %>% 
  mutate(rarA_truncation=ifelse(rarA_coverage < 90, 1, 0)) %>% 
  subset(select = -c(rarA_coverage)) %>%
  # consider any genome with at least one complete rarA as no rarA truncation
  arrange(strain, match(rarA_truncation, c(0, 1))) %>%
  distinct(strain, .keep_all = TRUE)

# rarA not identified in nucleotide BLAST search (i.e., rarA deletion) considered to be truncation
geno_rarA_truncation <- full_join(geno,rarA_truncation) %>%
   mutate(rarA_truncation=replace_na(rarA_truncation, 1))


```


``` {r rob truncation}

# rob truncation is defined as a genome with all rob copies with < 90% acid sequence coverage

rob_truncation <- rob_coverage %>% 
  mutate(rob_truncation=ifelse(rob_coverage < 90, 1, 0)) %>% 
  subset(select = -c(rob_coverage)) %>%
  # consider any genome with at least one complete rob as no rob truncation
  arrange(strain, match(rob_truncation, c(0, 1))) %>%
  distinct(strain, .keep_all = TRUE)

# rob not identified in nucleotide BLAST search (i.e., rob deletion) considered to be truncation
geno_rob_truncation <- full_join(geno,rob_truncation) %>%
   mutate(rob_truncation=replace_na(rob_truncation, 1))

```


``` {r oqxR truncation + mutations (amrfinder)}

# oqxR truncation is defined as a genome with all oqxR copies with < 90% acid sequence coverage

oqxR_truncation <- oqxR_coverage %>% 
  mutate(oqxR_truncation=ifelse(oqxR_coverage < 90, 1, 0)) %>% 
  subset(select = -c(oqxR_coverage)) %>%
  # consider any genome with at least one complete oqxR as no oqxR truncation
  arrange(strain, match(oqxR_truncation, c(0, 1))) %>%
  distinct(strain, .keep_all = TRUE)

# oqxR not identified in nucleotide BLAST search (i.e., oqxR deletion) considered to be truncation
geno_oqxR_truncation <- full_join(geno,oqxR_truncation) %>%
   mutate(oqxR_truncation=replace_na(oqxR_truncation, 1))

oqxR_mutation <- oqxR_mutation %>% 
  select(strain,`Gene symbol`) %>% 
  mutate(value = 1)  %>% 
  spread(`Gene symbol`, value,  fill = 0 ) 

# oqxR mutations not identified 
geno_oqxR_truncation_mutation <- full_join(geno_oqxR_truncation,oqxR_mutation) %>%
   mutate(oqxR_D95E=replace_na(oqxR_D95E, 0)) %>%
  mutate(oqxR_Q11L=replace_na(oqxR_Q11L, 0)) %>%
  mutate(oqxR_V113I=replace_na(oqxR_V113I, 0))

```



```{r combine geno+pheno}
#combining truncation from all oqxAB operon genes
geno_oqxABoperon_truncation <-Reduce(function(x,y) merge(x,y,by="strain",all=TRUE) ,list( geno_oqxR_truncation_mutation, geno_norB_truncation[, c("strain","norB_truncation")] ,geno_rarA_truncation[, c("strain","rarA_truncation")],geno_rob_truncation[, c("strain","rob_truncation")]))

cipro_antibiogram <- full_join(geno_oqxABoperon_truncation,pheno) %>%
  mutate(SRnwt = if_else(Resistance.phenotype=="I", "NWT", Resistance.phenotype)) %>%
  mutate(SRnwt = factor(SRnwt, levels=c("S","NWT","R")))

```

#Individual contribution of oqxAB negative regulator truncations/loss and oqxR mutations
```{r truncations - table with PPVs}

# Contribution of truncation on cipro susceptibility in genomes with no Flq resistance determinants

oqxR_truncation_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  group_by(oqxR_truncation) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
  rename(`Presence/absence`=oqxR_truncation)%>%
  mutate(Gene=c(rep("oqxR_truncation_loss",2)))%>%
  relocate(Gene, .before=`Presence/absence`)

norB_truncation_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  group_by(norB_truncation) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
  rename(`Presence/absence`=norB_truncation)%>%
  mutate(Gene=c(rep("norB_truncation_loss",2)))%>%
  relocate(Gene, .before=`Presence/absence`)

rarA_truncation_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  group_by(rarA_truncation) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
  rename(`Presence/absence`=rarA_truncation)%>%
  mutate(Gene=c(rep("rarA_truncation_loss",2)))%>%
  relocate(Gene, .before=`Presence/absence`)

rob_truncation_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  group_by(rob_truncation) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
  rename(`Presence/absence`=rob_truncation)%>%
  mutate(Gene=c(rep("rob_truncation_loss",2)))%>%
  relocate(Gene, .before=`Presence/absence`)


oqxR_D95E_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  group_by(oqxR_D95E) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
  rename(`Presence/absence`=oqxR_D95E)%>%
  mutate(Gene=c(rep("oqxR_D95E",2)))%>%
  relocate(Gene, .before=`Presence/absence`)

oqxR_V113I_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  group_by(oqxR_V113I) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
  rename(`Presence/absence`=oqxR_V113I)%>%
  mutate(Gene=c(rep("oqxR_V113I",2)))%>%
  relocate(Gene, .before=`Presence/absence`)



```



# Odds ratios, p-values, prop tests for each oqxAB neg reg defect
```{r}
noResDet <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0)

# Fishers test for oqxR_truncation copy vs. R phenotype
fisher.test(noResDet$oqxR_truncation, noResDet$resistant)
prop.test(table(noResDet$oqxR_truncation,noResDet$resistant)[2:1,2:1])

# Fishers test for norB_truncation copy vs. R phenotype
fisher.test(noResDet$norB_truncation, noResDet$resistant)
prop.test(table(noResDet$norB_truncation,noResDet$resistant)[2:1,2:1])

# Fishers test for rarA_truncation copy vs. R phenotype
fisher.test(noResDet$rarA_truncation, noResDet$resistant)
prop.test(table(noResDet$rarA_truncation,noResDet$resistant)[2:1,2:1])

# Fishers test for rob_truncation copy vs. R phenotype
fisher.test(noResDet$rob_truncation, noResDet$resistant)
prop.test(table(noResDet$rob_truncation,noResDet$resistant)[2:1,2:1])

# Fishers test for oqxR_D95E  vs. R phenotype
fisher.test(noResDet$oqxR_D95E, noResDet$resistant)
prop.test(table(noResDet$oqxR_D95E,noResDet$resistant)[2:1,2:1])

# Fishers test for oqxR_V113I  vs. R phenotype
fisher.test(noResDet$oqxR_V113I, noResDet$resistant)
prop.test(table(noResDet$oqxR_V113I,noResDet$resistant)[2:1,2:1])

```
#Summary contribution of oqxAB negative regulator truncations/loss and oqxR mutations
```{r}
# Any oqxAB negative regulator truncations/loss and oqxR mutations counted as the same

cipro_antibiogram <- cipro_antibiogram %>% 
  rowwise() %>%
mutate(oqxAB_neg_reg_loss = sum(oqxR_truncation, norB_truncation, rarA_truncation, rob_truncation, oqxR_D95E, oqxR_Q11L, oqxR_V113I, na.rm = T)) %>%
  mutate(oqxAB_neg_sum = if_else((oqxAB_neg_reg_loss>0), 1, 0))

noResDet <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0)

noResDet %>% filter(oqxAB_neg_sum==1) %>% nrow()

# Fishers test for >1 oqxB copy vs. R phenotype
fisher.test(noResDet$oqxAB_neg_sum, noResDet$resistant)
prop.test(table(noResDet$oqxAB_neg_sum,noResDet$resistant)[2:1,2:1])

```



```{r upset_plot MIC}

# Contribution of truncation on cipro susceptibility in genomes with no Flq resistance determinants

oqx_truncation_MIC <- cipro_antibiogram %>% 
  filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
  upset(
      intersect=c("oqxR_truncation","norB_truncation", "rarA_truncation", "rob_truncation", "oqxR_D95E", "oqxR_V113I"), # counts are converted to binary
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45),axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing WT/nonWT
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
                 scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_colour_manual(values = res_colours) + 
              scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:9), labels = function(x) round(as.numeric(x), digits = 3), limits=c(2^-5,2^9)) +
              theme(legend.position = "none", axis.title=element_text(size=8))  +
              ggtitle("Genomes with no Flq resistance determinants - MIC")
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

oqx_truncation_MIC
#ggsave("figs/FigSX_oqxAB_truncation_MIC.pdf", width=8, height=7)
#ggsave("figs/FigSX_oqxAB_truncation_MIC.png", width=8, height=7)
```

```{r upset_plot DD}
oqx_truncation_DD <- cipro_antibiogram %>% 
  filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
  filter(Laboratory.Typing.Method == "Disk diffusion") %>%
  upset(
      intersect=c("oqxR_truncation","norB_truncation", "rarA_truncation", "rob_truncation","oqxR_D95E", "oqxR_V113I"), # counts are converted to binary
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45), axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
               scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            theme(legend.position = "none", axis.title=element_text(size=8)),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with disk diffusion data
        'Zone diameter (mm)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
              geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_color_manual(values = c("I"="grey", "R"="IndianRed", "S"="LightBlue")) + 
              guides(size = "none") +
              labs(linetype = "CLSI/EUCAST", colour = "Phenotype") +
              ggtitle("Genomes with no Flq resistance determinants - \n
                      Disk diffusion") +
              theme(axis.title=element_text(size=8)) + ylim(5,50)
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

oqx_truncation_DD

#ggsave("figs/FigSX_oqxAB_truncation_DD.pdf", width=8, height=7)
#ggsave("figs/FigSX_oqxAB_truncation_DD.png", width=8, height=7)
```

# Copy number

```{r oqxA_MIC}
oqxA_MIC_noGenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
    filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>% 
    ggplot(aes(x=factor(oqxA_copy), y=as.numeric(Measurement))) + 
    geom_violin() + 
    geom_count(aes(colour = SRnwt)) + 
    geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
    geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
    scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:9), labels = function(x) round(as.numeric(x), digits = 3)) +
    scale_color_manual(values = res_colours) + 
    theme_light() + 
    labs(y="MIC (mg/L)", x="", col="Phenotype", 
         title="oqxA copy number vs phenotype",
         subtitle="(in absence of other Flq res determinants)")



oqxA_MICpheno_noGenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
    filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
    ggplot(aes(x=factor(oqxA_copy), fill=SRnwt)) + 
    geom_bar(stat='count', position='fill') + 
    scale_fill_manual(values = res_colours) + 
    scale_y_continuous(labels=scales::percent_format()) +
    geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), size=2) +
    theme_light() + 
    labs(y="% Phenotype", x="oqxA copy number") + 
    theme(legend.position="none", strip.background = element_blank(), strip.text = element_blank())

oqxA_MIC_noGenes / oqxA_MICpheno_noGenes + plot_layout(heights=c(2,1))
#ggsave("figs/FigSX_oqxA_copy_MIC_pheno.pdf", width=6, height=4)
#ggsave("figs/FigSX_oqxA_copy_MIC_pheno.png", width=6, height=4)
```

``` {r oqxA_DD}

# DD distribution for copy number, in absence of genes
oqxA_DD_nogenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
    filter(Laboratory.Typing.Method=="Disk diffusion") %>% 
    ggplot(aes(x=factor(oqxA_copy), y=as.numeric(Measurement))) + 
    geom_violin() + 
    geom_count(aes(colour = SRnwt)) + 
    geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
    geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
    scale_colour_manual(values = res_colours) + 
    theme_light() + 
    labs(y="Disk zone (mm)", x="", col="Phenotype",
         title="oqxA copy number vs phenotype",
         subtitle="(in absence of other Flq res determinants)")

oqxA_DDpheno_nogenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
    filter(Laboratory.Typing.Method=="Disk diffusion") %>%
    ggplot(aes(x=factor(oqxA_copy), fill=SRnwt)) + 
    geom_bar(stat='count', position='fill') + 
    scale_fill_manual(values = res_colours) +  
    scale_y_continuous(labels=scales::percent_format()) +
    geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), size=2) +
    theme_light() + 
    labs(y="% Phenotype", x="oqxA copy number") + 
    theme(legend.position="none", strip.background = element_blank(), strip.text = element_blank())

oqxA_DD_nogenes / oqxA_DDpheno_nogenes + plot_layout(heights=c(2,1))

#ggsave("figs/FigSX_oqxA_copy_DD_pheno.pdf", width=6, height=4)
#ggsave("figs/FigSX_oqxA_copy_DD_pheno.png", width=6, height=4)
```



```{r oqxB_MIC}
oqxB_MIC_noGenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
    filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>% 
    ggplot(aes(x=factor(oqxB_copy), y=as.numeric(Measurement))) + 
    geom_violin() + 
    geom_count(aes(colour = SRnwt)) + 
    geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
    geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
    scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:9), labels = function(x) round(as.numeric(x), digits = 3)) +
    scale_color_manual(values = res_colours) + 
    theme_light() + 
    labs(y="MIC (mg/L)", x="", col="Phenotype", 
         title="oqxB copy number vs phenotype",
         subtitle="(in absence of other Flq res determinants)")



oqxB_MICpheno_noGenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
    filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
    ggplot(aes(x=factor(oqxB_copy), fill=SRnwt)) + 
    geom_bar(stat='count', position='fill') + 
    scale_fill_manual(values = res_colours) + 
    scale_y_continuous(labels=scales::percent_format()) +
    geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), size=2) +
    theme_light() + 
    labs(y="% Phenotype", x="oqxB copy number") + 
    theme(legend.position="none", strip.background = element_blank(), strip.text = element_blank())

oqxB_MIC_noGenes / oqxB_MICpheno_noGenes + plot_layout(heights=c(2,1))
#ggsave("figs/FigSX_oqxB_copy_MIC_pheno.pdf", width=6, height=4)
#ggsave("figs/FigSX_oqxB_copy_MIC_pheno.png", width=6, height=4)
```

``` {r oqxB_DD}

# DD distribution for copy number, in absence of genes
oqxB_DD_nogenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
    filter(Laboratory.Typing.Method=="Disk diffusion") %>% 
    ggplot(aes(x=factor(oqxB_copy), y=as.numeric(Measurement))) + 
    geom_violin() + 
    geom_count(aes(colour = SRnwt)) + 
    geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
    geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
    scale_colour_manual(values = res_colours) + 
    theme_light() + 
    labs(y="Disk zone (mm)", x="", col="Phenotype",
         title="oqxB copy number vs phenotype",
         subtitle="(in absence of other Flq res determinants)")

oqxB_DDpheno_nogenes <- cipro_antibiogram %>%
    filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
    filter(Laboratory.Typing.Method=="Disk diffusion") %>%
    ggplot(aes(x=factor(oqxB_copy), fill=SRnwt)) + 
    geom_bar(stat='count', position='fill') + 
    scale_fill_manual(values = res_colours) +  
    scale_y_continuous(labels=scales::percent_format()) +
    geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), size=2) +
    theme_light() + 
    labs(y="% Phenotype", x="oqxB copy number") + 
    theme(legend.position="none", strip.background = element_blank(), strip.text = element_blank())

oqxB_DD_nogenes / oqxB_DDpheno_nogenes + plot_layout(heights=c(2,1))

#ggsave("figs/FigSX_oqxB_copy_DD_pheno.pdf", width=6, height=4)
#ggsave("figs/FigSX_oqxB_copy_DD_pheno.png", width=6, height=4)
```

```{r oqxab_copy preparing upset plot data}
cipro_antibiogram <- cipro_antibiogram %>%
    mutate(oqxA_0 = if_else((oqxA_copy==0), 1, 0))%>%  
    mutate(oqxA_1 = if_else((oqxA_copy==1), 1, 0))%>%
    mutate(oqxA_2 = if_else((oqxA_copy>1), 1, 0))%>%
    mutate(oqxB_0 = if_else((oqxB_copy==0), 1, 0))%>%  
    mutate(oqxB_1 = if_else((oqxB_copy==1), 1, 0))%>%
    mutate(oqxB_2 = if_else((oqxB_copy>1), 1, 0))
```


```{r oqxab_copy_upset_plot MIC}

oqx_truncation_MIC <- cipro_antibiogram %>% 
  filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
  filter(grepl('MIC.*$', Laboratory.Typing.Method)) %>%
  upset(
      intersect=c("oqxA_0","oqxA_1", "oqxA_2","oqxB_0", "oqxB_1", "oqxB_2"), # counts are converted to binary
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45),axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing WT/nonWT
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
                 scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 1), linetype = 2, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 0.25), linetype = 1, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_colour_manual(values = res_colours) + 
              scale_y_continuous(trans = log2_trans(), breaks = 2^(-5:9), labels = function(x) round(as.numeric(x), digits = 3), limits=c(2^-5,2^9)) +
              theme(legend.position = "none", axis.title=element_text(size=8))  +
              ggtitle("Genomes with no Flq resistance determinants - MIC")
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

oqx_truncation_MIC
#ggsave("figs/FigSX_oqxAB_copy_MIC.pdf", width=8, height=7)
#ggsave("figs/FigSX_oqxAB_copy_MIC.png", width=8, height=7)
```

```{r oqxab_copy_upset_plot DD}
oqxAB_copy_DD <- cipro_antibiogram %>% 
  filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>%
  filter(Laboratory.Typing.Method == "Disk diffusion") %>%
  upset(
      intersect=c("oqxA_0","oqxA_1", "oqxA_2","oqxB_0", "oqxB_1", "oqxB_2"), # counts are converted to binary
      set_sizes=(upset_set_size() +
                    theme(axis.text.x=element_text(angle=45), axis.ticks.x=element_line())),
      base_annotations=list( # add stacked barplot showing S/nonS
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=SRnwt),
          geom=list(
            geom_bar(stat='count', position='fill'), 
               scale_fill_manual(values = res_colours),
            scale_y_continuous(labels=scales::percent_format()),
            theme(legend.position = "none", axis.title=element_text(size=8)),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with disk diffusion data
        'Zone diameter (mm)'=(
            ggplot(mapping=aes(y=as.numeric(Measurement))) +
              geom_hline(aes(yintercept = 21.0), linetype = 1, alpha = 0.6, color = "black") +
              geom_hline(aes(yintercept = 25.0), linetype = 2, alpha = 0.6, color = "black") +
              geom_violin()+ 
              geom_count(aes(colour = SRnwt))+
               scale_color_manual(values = c("I"="grey", "R"="IndianRed", "S"="LightBlue")) + 
              guides(size = "none") +
              labs(linetype = "CLSI/EUCAST", colour = "Phenotype") +
              ggtitle("Genomes with no Flq resistance determinants - \n
                      Disk diffusion") +
              theme(axis.title=element_text(size=8)) + ylim(5,50)
        )
      )
    ) + patchwork::plot_layout(heights=c(3,1,1)) # relative heights of plotting areas

oqxAB_copy_DD

#ggsave("figs/FigSX_oqxAB_copy_DD.pdf", width=8, height=7)
#ggsave("figs/FigSX_oqxAB_copy_DD.png", width=8, height=7)
```

```{r extra_copy_oqxAB_PPV_table}
oqxA_copy_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  filter(oqxA_1 ==1 | oqxA_2 ==1)%>% 
  group_by(oqxA_2) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
    rename(`Presence/absence`=oqxA_2)%>%
  mutate(Gene=c(rep("extra_oqxA_copy",2)))%>%
  relocate(Gene, .before=`Presence/absence`)

noResDet <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0)
# Fishers test for >1 oqxA copy vs. R phenotype
fisher.test(noResDet$oqxA_2, noResDet$resistant)
prop.test(table(noResDet$oqxA_2,noResDet$resistant)[2:1,2:1])

#write.table(oqxA_copy_solo, "oqxA_copy_solo_PPV.tsv", row.names=FALSE, sep="\t")

oqxB_copy_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
    filter(oqxB_1 ==1 | oqxB_2 ==1)%>% 
  group_by(oqxB_2) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
    rename(`Presence/absence`=oqxB_2)%>%
  mutate(Gene=c(rep("extra_oqxB_copy",2)))%>%
  relocate(Gene, .before=`Presence/absence`)

# Fishers test for >1 oqxB copy vs. R phenotype
fisher.test(noResDet$oqxB_2, noResDet$resistant)
prop.test(table(noResDet$oqxB_2,noResDet$resistant)[2:1,2:1])

#write.table(oqxB_copy_solo, "oqxB_copy_solo_PPV.tsv", row.names=FALSE, sep="\t")
```

```{r loss_oqxAB_PPV_table}
oqxA_loss_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  filter(oqxA_1 ==1 | oqxA_0 ==1)%>% 
  group_by(oqxA_0) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
    rename(`Presence/absence`=oqxA_0)%>%
  mutate(Gene=c(rep("oqxA_loss",2)))%>%
  relocate(Gene, .before=`Presence/absence`)

noResDet <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0)
# Fishers test for >1 oqxA copy vs. R phenotype
fisher.test(noResDet$oqxA_2, noResDet$resistant)
prop.test(table(noResDet$oqxA_2,noResDet$resistant)[2:1,2:1])

oqxB_loss_solo <- cipro_antibiogram %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
    filter(oqxB_1 ==1 | oqxB_0 ==1)%>% 
  group_by(oqxB_0) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) %>%
    rename(`Presence/absence`=oqxB_0)%>%
  mutate(Gene=c(rep("oqxB_loss",2)))%>%
  relocate(Gene, .before=`Presence/absence`)

# Fishers test for >1 oqxB copy vs. R phenotype
fisher.test(noResDet$oqxB_2, noResDet$resistant)
prop.test(table(noResDet$oqxB_2,noResDet$resistant)[2:1,2:1])


```

#Table S10
```{r}
# Table S10- Contribution of solo truncations/mutations

oqxAB_operon_truncation_solo <- 
rbind(oqxR_truncation_solo, norB_truncation_solo, rarA_truncation_solo, rob_truncation_solo, oqxR_D95E_solo, oqxR_V113I_solo, oqxA_copy_solo, oqxB_copy_solo, oqxA_loss_solo, oqxB_loss_solo)

oqxAB_operon_truncation_solo

write.table(oqxAB_operon_truncation_solo, "tables/TableS10_oqxAB_contribution_PPV.tsv", row.names=FALSE, sep="\t")
```



#Insertion sequences
```{r}
#exclude genomes with no or multiple oqxAB copies

cipro_antibiogram_IS <-cipro_antibiogram %>% filter(oqxA_copy==1 & oqxB_copy==1)

cipro_antibiogram_IS <- left_join(cipro_antibiogram_IS,oqxab_IS_cipro %>% select(strain,gene)) %>%
   mutate(oqxa_IS=ifelse(!is.na(gene), 1, 0))

oqxa_IS_solo <- cipro_antibiogram_IS %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0) %>% 
  group_by(oqxa_IS) %>% 
  summarise(N=n(), S=sum(Resistance.phenotype=="S"), I=sum(Resistance.phenotype=="I"), R=sum(resistant), NWT=sum(nonWT_binary), R_PPV=mean(resistant)*100, NWT_PPV=mean(nonWT_binary)*100) 

#write.table(oqxa_IS_solo, "oqxa_IS_solo_PPV.tsv", row.names=FALSE, sep="\t")

noResDet <- cipro_antibiogram_IS %>% filter(QRDR_mutations==0 & aac6==0 & acquired_genes==0)

noResDet %>% filter(oqxa_IS==1) %>% nrow()

# Fishers test for >1 oqxB copy vs. R phenotype
fisher.test(noResDet$oqxa_IS, noResDet$resistant)
prop.test(table(noResDet$oqxa_IS,noResDet$resistant)[2:1,2:1])

```

# Unexplained resistance

```{r}
# number R with no res determinants
noResDet_R <- cipro_antibiogram %>% filter(Resistance.phenotype=="R" & QRDR_mutations==0 & acquired_genes==0 & aac6==0)

noResDet_R %>% filter(grepl('OmpK35', Omp_mutations))%>% nrow()

noResDet_R %>% filter(grepl('OmpK36', Omp_mutations))%>% nrow()

noResDet_R %>% filter(oqxA_2==1)%>% nrow()

noResDet_R %>% filter(oqxB_2==1)%>% nrow()

noResDet_R %>% filter(oqxAB_neg_sum==1)%>% nrow()

#no porin, multiple copies of oqxAB, or any oqxAB neg reg truncation/loss 
noResDet_R %>% filter(Omp_mutations=="-" & oqxA_2==0 & oqxB_2==0 & oqxAB_neg_sum==0)%>% nrow()

noResDet_R_nooqx_porin_defect <- noResDet_R %>% filter(Omp_mutations=="-" & oqxA_2==0 & oqxB_2==0 & oqxAB_neg_sum==0)

dd_atu <- noResDet_R_nooqx_porin_defect %>% filter(Resistance.phenotype!="R") %>% filter(Laboratory.Typing.Method == "Disk diffusion") %>% filter(as.numeric(Measurement) >21 & as.numeric(Measurement)<24) 

mic_atu <- noResDet_R_nooqx_porin_defect %>% filter(Resistance.phenotype!="R") %>% filter(Laboratory.Typing.Method != "Disk diffusion") %>% filter(as.numeric(Measurement) ==0.5) 

atu <- bind_rows(mic_atu, mic_atu) 

```

# Unexplained susceptibility

```{r}
# number S with mutations
S_mut <- cipro_antibiogram %>% filter(Resistance.phenotype=="S" & ((QRDR_mutations==1 & Flq_mutations!="GyrA-87G"& Flq_mutations!="GyrA-87H") | (acquired_genes==1 & !grepl('qnrB1.', Flq_acquired)))) 

# num with loss of oqxAB
S_mut %>% filter(oqxA_0==1 & oqxB_0==1) %>% nrow()

S_mut %>% filter(oqxA_0==1) %>% nrow()

S_mut %>% filter(oqxB_0==1) %>% nrow()


```