---
title: "Pathogenwatch analysis"
author: "Kara Tsang, Kat Holt"
date: "Last compiled on `r format(Sys.time(), '%d/%m/%y')`"
output: 
  html_document: 
    df_print: paged
    fig_width: 9
    fig_height: 8
    mode: selfcontained
    highlight: pygments
    theme: 
      bootswatch: flatly
      code_font:
        google: "JetBrains Mono"
editor_options: 
  chunk_output_type: inline
---

```{=html}
<style type="text/css">
.main-container {
  max-width: 1600px !important;
  margin-left: auto;
  margin-right: auto;
}
</style>
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = 'png', echo = TRUE, message = FALSE, warning = FALSE,
                      cache = FALSE, fig.align = 'center')
knitr::opts_knit$set(root.dir = getwd())

# Data wrangling
library(tidyverse)
library(dplyr)
library(patchwork)
library(countrycode)
library(ggpubr)
library(ggrepel)
library(pheatmap)
```

# load data
``` {r}

#Pathogenwatch data downloaded Nov 7, 2024

kpn_kleborate_pathogenwatch <- read.delim(file = "Klebsiella pneumoniae__kleborate.csv.gz", header = TRUE, sep = ",")
kpn_metadata_pathogenwatch <- read.delim(file = "Klebsiella pneumoniae__metadata.csv.gz", header = TRUE, sep = ",")

# K. africana doesn't have kleborate results
#ka_kleborate_pathogenwatch <- read.delim(file = "Klebsiella africana__kleborate.csv.gz", header = TRUE, sep = ",")
#ka_metadata_pathogenwatch <- read.delim(file = "Klebsiella africana__metadata.csv.gz", header = TRUE, sep = ",")

# K quasipneumoniae doesn't have kleborate results
#kqp_kleborate_pathogenwatch <- read.delim(file = "Klebsiella quasipneumoniae__kleborate.csv.gz", header = TRUE, sep = ",")
#kq_metadata_pathogenwatch <- read.delim(file = "Klebsiella quasipneumoniae__metadata.csv.gz", header = TRUE, sep = ",")

# K quasivariicola doesn't have kleborate results
#kqv_kleborate_pathogenwatch <- read.delim(file = "Klebsiella quasivariicola__kleborate.csv.gz", header = TRUE, sep = ",")
#kqv_metadata_pathogenwatch <- read.delim(file = "Klebsiella quasivariicola__metadata.csv.gz", header = TRUE, sep = ",")

# K variicola doesn't have kleborate results
#kv_kleborate_pathogenwatch <- read.delim(file = "Klebsiella variicola__kleborate.csv.gz", header = TRUE, sep = ",")
#kv_metadata_pathogenwatch <- read.delim(file = "Klebsiella variicola__metadata.csv.gz", header = TRUE, sep = ",")

pathogenwatch <- left_join(kpn_kleborate_pathogenwatch, kpn_metadata_pathogenwatch, by = c("Genome.ID" = "id")) %>% distinct() %>% select(Flq_acquired, Flq_mutations, AGly_acquired, year, Country, ST, species)
```
# genotype labels and colours
```{r labels_colours}
profile_labels <- paste(
  paste0(c("0^", "0", "0", "1", "1", ">1", "0", "0", "0", ">0"), rep(" QRDR,", 10)),
  c("0 PMQR,", "0 PMQR,", "qnrB1,", "0 PMQR,", "0 PMQR,", "0 PMQR", "1 PMQR^,", "1 PMQR,", ">1 PMQR", ">0 PMQR"),
  c("aac6-", "aac6+", "aac6-", "aac6-", "aac6+", "", "aac6-", "aac6+", "", "")
)
names(profile_labels)<-0:9

profile_colours <- c("0" = "#9DDFFE", "1"="#1A83B6", "2"="#98d3a4", "3"="#fbe8a5", "4"="#ecb16f", "5"="#f17c1b", "6"="#fdc0c0", "7"="#e88f8f", "8"="#b92020", "9"="#59124b")
```
## count mutations and genes
``` {r}

pathogenwatch <- pathogenwatch %>%
  mutate(aac6 = ifelse(grepl("aac\\(6\\'\\)-Ib-cr", AGly_acquired, ignore.case = T), 1, 0)) %>%
  mutate(GyrA_mutations = str_count(Flq_mutations, "GyrA")) %>%
  mutate(ParC_mutations = str_count(Flq_mutations, "ParC")) %>% 
  mutate(QRDR_mutations = GyrA_mutations+ParC_mutations) %>%
  mutate(qnr = str_count(Flq_acquired, "qnr")) %>%
  mutate(qep = str_count(Flq_acquired, "qep")) %>%
  mutate(acquired_genes = qnr+qep) %>%
  mutate(`GyrA-83` = str_count(Flq_mutations, "GyrA-83")) %>%
  mutate(`GyrA-87` = str_count(Flq_mutations, "GyrA-87")) %>%
  mutate(`ParC-80` = str_count(Flq_mutations, "ParC-80")) %>% 
  mutate(`ParC-84` = str_count(Flq_mutations, "ParC-84")) %>%
  mutate(wt_QRDR = if_else(QRDR_mutations==0, 1, 0)) 

```


# rules category

```{r rules_category}


pathogenwatch <- pathogenwatch %>% 
  mutate(QRDR_mutations_exclude = str_count(Flq_mutations, "GyrA-87G") + str_count(Flq_mutations, "GyrA-87H")) %>% 
  mutate(PMQR_exclude = str_count(Flq_acquired, "qnrB1.v1") + str_count(Flq_acquired, "qnrB1.v2")) %>%
  mutate(rules_category = case_when(
    QRDR_mutations==0 & acquired_genes==0 & aac6==0 ~ 0, # no determinants -> WT S
    QRDR_mutations==1 & acquired_genes==0 & aac6==0 & QRDR_mutations_exclude==1 ~ 0, # non-associated QRDR solo -> WT S
    QRDR_mutations==0 & acquired_genes==0 & aac6==1 ~ 1, # aac6 solo -> WT S
    QRDR_mutations==0 & acquired_genes==1 & aac6==0 & PMQR_exclude==1 ~ 2, # non-associated PMQR solo -> NWT I
    QRDR_mutations==1 & acquired_genes==0 & aac6==0 & QRDR_mutations_exclude==0 ~ 3, # 1 QRDR, except those not associated -> NWT R
    QRDR_mutations==1 & acquired_genes==0 & aac6==1 ~ 4, # 1 QRDR + aac6 -> NWT R
    QRDR_mutations>=2 & acquired_genes==0 ~ 5, # ≥2 QRDR -> NWT R
    QRDR_mutations==0 & acquired_genes==1 & aac6==0 & PMQR_exclude==0 ~ 6, # 1 PMQR, except those not associated -> NWT R
    QRDR_mutations==0 & acquired_genes==1 & aac6==1 ~ 7, # 1 PMQR + aac6
    QRDR_mutations==0 & acquired_genes>=2 ~ 8, # ≥2 PMQR -> NWT R
    QRDR_mutations>=1 & acquired_genes>=1 ~ 9, # QRDR + PMQR -> NWT R
    TRUE ~ NA
  ))

# check all are assigned
table(pathogenwatch$rules_category)
table(is.na(pathogenwatch$rules_category))
pathogenwatch %>% filter(is.na(rules_category)) %>% select(Flq_mutations, Flq_acquired)

# assign S/R predictions to categories
pathogenwatch <- pathogenwatch %>% 
  mutate(predSIR = if_else(rules_category<=2, "S", "R")) %>%
  mutate(predSIR = replace(predSIR, rules_category==2, "I")) %>%
  mutate(predSIR = factor(predSIR, levels=c("S", "I", "R"))) %>%
  mutate(predR = if_else(rules_category<=2, "S", "R")) %>%
  mutate(predR = factor(predR, levels=c("S", "R"))) %>%
  mutate(predNWT = if_else(rules_category<2, "WT", "NWT")) %>%
  mutate(predNWT = factor(predNWT, levels=c("WT", "NWT")))

table(pathogenwatch$rules_category, pathogenwatch$predR)
table(pathogenwatch$rules_category, pathogenwatch$predSIR)
table(pathogenwatch$rules_category, pathogenwatch$predNWT)

PW <- pathogenwatch

```


# numbers for text
``` {r}
# total genomes
nrow(PW)

# restrict to those with year & Country
PW <- PW %>% filter(Country !="missing" & Country !="Other (International Space Station)") %>% filter(year>=2000)

# number included
nrow(PW)
length(unique(PW$Country))

ST_freq <- PW %>% group_by(ST, species) %>% summarise(N=n()) %>% arrange(-N)

ST_freq[1:10,]

ST_freq %>% filter(N>=100) %>% pull(N) %>% sum()

# check species - these are all Kp
table(ST_freq$species[ST_freq$N>=100])

# Kp R%
spp_count <- PW %>% group_by(species) %>% summarise(Nspp=n())
PW %>% group_by(species, predR) %>% count() %>% right_join(spp_count) %>% mutate(pc=n/Nspp)
PW %>% group_by(species, predNWT) %>% count() %>% right_join(spp_count) %>% mutate(pc=n/Nspp)

# species vs variant
PW %>% group_by(species, QRDR_mutations) %>% count()%>% right_join(spp_count) %>% mutate(pc=n/Nspp) %>% mutate(pc_inv=1-pc)
PW %>% group_by(species, acquired_genes) %>% count()%>% right_join(spp_count) %>% mutate(pc=n/Nspp) %>% mutate(pc_inv=1-pc)
PW %>% group_by(species, aac6) %>% count()%>% right_join(spp_count) %>% mutate(pc=n/Nspp) %>% mutate(pc_inv=1-pc)

PW %>% group_by(ST, species, predR) %>% count() %>% right_join(ST_freq) %>% mutate(pc=n/N) %>% filter(ST=="ST347")
PW %>% group_by(ST, species, rules_category) %>% count() %>% right_join(ST_freq) %>% mutate(pc=n/N) %>% filter(ST=="ST347")

PW %>% group_by(ST, species, rules_category) %>% count() %>% right_join(ST_freq) %>% mutate(pc=n/N) %>% filter(rules_category==0) %>% arrange(-N)
```

# fig - ST vs determinants, with ST freq
``` {r}

# clean up characters
PW <- PW %>% 
  mutate(Flq_acquired = gsub("[\\^\\*\\?]", "", Flq_acquired))

ST_pred <- PW %>% group_by(ST, predR) %>% count() %>% right_join(ST_freq) %>% filter(predR=="R") %>% mutate(pc=n/N)

ST_cat <- PW %>% group_by(ST, rules_category) %>% count() %>% right_join(ST_freq) %>% mutate(pc=n/N)

``` 


# kp plot
```{r}
PMQR_freq_kp <- PW %>%
    filter(species=="Klebsiella pneumoniae") %>%
    select(Flq_acquired) %>%
    separate_longer_delim(Flq_acquired, delim = ";") %>%
    filter(Flq_acquired !="") %>%
    filter(Flq_acquired !="-") %>%
    group_by(Flq_acquired) %>% 
    count() %>% arrange(-n)

QRDR_freq_kp <- PW %>%
    filter(species=="Klebsiella pneumoniae") %>%
    select(Flq_mutations) %>%
    separate_longer_delim(Flq_mutations, delim = ";") %>%
    filter(Flq_mutations !="") %>%
    filter(Flq_mutations !="-") %>%
    group_by(Flq_mutations) %>% 
    count() %>% arrange(-n)


PW_dat_variant_kp <- PW %>% 
  filter(species=="Klebsiella pneumoniae") %>%
  mutate(Flq_acquired_mut=paste(Flq_acquired, Flq_mutations, sep = ";")) %>%
  mutate(aac6_label=if_else(aac6==1, "aac6", "")) %>%
  mutate(variant=paste(Flq_acquired_mut, aac6_label, sep = ";")) %>%
  filter(!is.na(ST)) %>%
  mutate(variant = gsub(' ', '', variant)) %>%
  select(ST, variant) %>%
  separate_longer_delim(variant, delim = ";") %>%
  filter(variant!="-") %>%
  filter(variant!="")

# ST vs variant, for STs with N>=100 (Kp)
PW_dat_kp <- PW_dat_variant_kp %>%
  filter(ST %in% ST_freq$ST[ST_freq$N>=100]) %>%
  group_by(ST, variant) %>% count() %>%
  left_join(ST_freq, by="ST") %>%
  mutate(pc=n/N)

# order of variants
variant_order_kp <- c("aac6", 
                   PMQR_freq_kp$Flq_acquired[PMQR_freq_kp$Flq_acquired %in% PW_dat_kp$variant],
                   QRDR_freq_kp$Flq_mutations[QRDR_freq_kp$Flq_mutations %in% PW_dat_kp$variant]
                  )
```

# variants vs ST heatmap
``` {r}
ST_cat_plot_kp <- ST_cat %>%
  ggplot(aes(x=ST, y=pc, fill=as.character(rules_category))) +
  geom_col() + 
  xlim(ST_freq$ST[ST_freq$N>100]) +
  theme_bw() + 
  scale_fill_manual(values=profile_colours, labels=profile_labels)+
  theme(axis.text.x = element_text(size=7, angle=60, hjust=1),
        axis.title.x = element_text(size=9),
        axis.title.y = element_text(angle=0, size=9),
        plot.margin = margin(0, 10, 0, 0, "pt"),
        legend.position="bottom",
        legend.text=element_text(size=7), 
        legend.key.size = unit(3,'mm'),
        legend.title=element_text(size=9)) + 
  labs(x="ST", y="Genotype\nprofile", fill="") + 
  guides(fill = guide_legend(nrow=3))

variant_heatmap_kp <- PW_dat_kp %>%
  ggplot(aes(x=ST, y=variant, fill=pc)) +
  geom_tile() + 
  xlim(ST_freq$ST[ST_freq$N>100]) +
  scale_y_discrete(limits=rev(variant_order_kp)) +
  theme_bw() + 
  theme(axis.text.x = element_text(size=7, angle=60, hjust=1), 
        axis.text.y = element_text(size=7),
        axis.title.x = element_text(size=9),
        axis.title.y = element_text(size=9, angle=0),
        legend.title = element_text(size=9),
        plot.margin = margin(0, 0, 0, 0, "pt")) + 
  labs(fill="Variant freq within ST", y="Variant", x="ST") + 
  scale_fill_gradient(low="#fbe8a5", high="navy") + 
  theme(legend.position="bottom")

var_freq_plot_kp <- PW_dat_variant_kp %>% 
  group_by(variant) %>% count() %>% 
  ggplot(aes(y=variant, x=n)) + 
  geom_col(fill="navy") + 
  theme_bw() + 
  geom_text(aes(y=variant, x=n+100, label=n), size=2.5, adj=00) +
  theme(axis.text.x = element_text(size=8, angle=60, hjust=1),
        axis.title.x = element_text(size=9),
        plot.margin = margin(0, 0, 0, 0, "pt"), 
        axis.text.y=element_blank()) + 
  ylim(rev(variant_order_kp)) + 
  xlim(c(0,35000)) + 
  labs(x="Variant freq", y="")

ST_freq_plot_kp <- ST_freq %>%
  ggplot(aes(x=ST, y=N)) +
  geom_col(fill="navy") + 
   xlim(ST_freq$ST[ST_freq$N>100]) +
  theme_bw() + 
  theme(axis.text.x = element_blank(), 
        axis.title.y = element_text(angle=0, size=9),
        plot.margin = margin(0, 10, 0, 0, "pt")) + 
  labs(x="", y="\nST\nfreq")

ST_predR_plot_kp <- ST_pred %>%
  ggplot(aes(x=ST, y=pc)) +
  geom_col(fill="navy") + 
  xlim(ST_freq$ST[ST_freq$N>100]) +
  theme_bw() + 
  theme(axis.text.x = element_blank(), 
        axis.title.y = element_text(angle=0, size=9),
        plot.margin = margin(0, 10, 0, 0, "pt")) + 
  labs(x="", y="Predicted\nR")
```

# Figure S11 - PW determinants vs ST

``` {r}
(ST_freq_plot_kp + plot_spacer() + plot_layout(widths=c(6,1))) /
    (variant_heatmap_kp + var_freq_plot_kp + plot_layout(widths=c(6,1))) /
    (ST_predR_plot_kp + plot_spacer() + plot_layout(widths=c(6,1))) /
    (ST_cat_plot_kp + plot_spacer() + plot_layout(widths=c(6,1))) + plot_layout(heights=c(1,8,1,1))

ggsave(height=9, width=7, file="PW_geneSTheatmapKp.png")

ggsave(height=9, width=7, file="PW_geneSTheatmapKp.pdf")
 
```


``` {r}
# country count
PW_country_count <- PW %>% filter(species=="Klebsiella pneumoniae") %>% group_by(Country) %>% summarise(countryN=n())

kp_profile_year <- PW %>% filter(species=="Klebsiella pneumoniae") %>% 
  group_by(year, rules_category) %>% count() %>%
  ggplot(aes(x=year, y=n, fill=as.character(rules_category))) + 
  geom_bar(stat="identity", position="fill") + 
 # coord_flip() +
  scale_fill_manual(values=profile_colours, labels=profile_labels) + 
  theme_bw() + 
  scale_y_continuous(labels = scales::percent)+
  labs(x="Year", y="Percentage of genomes", fill="Genotype\nprofile")

annual_multires <- PW %>% filter(species=="Klebsiella pneumoniae") %>% group_by(year) %>% summarise(Both=mean(rules_category==9), QRDR=mean(QRDR_mutations>0), PMQR=mean(acquired_genes>0)) %>% pivot_longer(cols=Both:PMQR)

annual_multires

annual_multires %>% filter(name=="Both") %>% arrange(-year)

annual_multires_plot <- annual_multires %>% 
  ggplot(aes(x=year, y=value*100, col=name)) + geom_point() + geom_line() +
  scale_colour_manual(values=c(QRDR="#ecb16f", PMQR="#b92020", Both="#59124b")) + 
  theme_bw() + 
  labs(x="Year", y="Percentage of genomes", col="Determinants")

```


# MIC prediction vs. year
```{r MIC_year}

MIC_profile_colours <- c("<=0.25 [0.25-0.25] mg/L" = "#add8e6", "0.25 [0.25-0.5] mg/L"="#72bcd4", "0.5 [0.5-1] mg/L"="#78638a", "1 [1-2] mg/L"="#fadbd8", "2 [1-2] mg/L"="#f1948a", "2 [2-4] mg/L"="#C64343", ">=4 [4-4] mg/L"="#4A0404")

MIC_pathogenwatch <- PW %>% mutate(MIC_prediction=case_when(rules_category==0 ~ "<=0.25 [0.25-0.25] mg/L",
                                  rules_category==1 ~ "0.25 [0.25-0.5] mg/L",
                                  rules_category==2 ~ "0.5 [0.5-1] mg/L",
                                  rules_category==3 ~ "1 [1-2] mg/L",
                                  rules_category==4 ~ "2 [1-2] mg/L",
                                  rules_category==5 ~ "2 [2-4] mg/L",
                                  rules_category==6 ~ "1 [1-2] mg/L",
                                  rules_category==7 ~ "2 [1-2] mg/L",
                                  rules_category==8 ~ "2 [2-4] mg/L",
                                  rules_category==9 ~ ">=4 [4-4] mg/L",
                                  TRUE ~ NA
                                  ))



MIC_pathogenwatch$MIC_prediction <- factor(MIC_pathogenwatch$MIC_prediction, levels=c("<=0.25 [0.25-0.25] mg/L", "0.25 [0.25-0.5] mg/L", "0.5 [0.5-1] mg/L", "1 [1-2] mg/L", "2 [1-2] mg/L", "2 [2-4] mg/L",">=4 [4-4] mg/L"))


MIC_year_pathogenwatch_plot <- MIC_pathogenwatch  %>% 
  filter(species=="Klebsiella pneumoniae")  %>% 
   group_by(year, MIC_prediction) %>% count() %>%
  ggplot(aes(fill=as.factor(MIC_prediction), y=n, x=year)) + 
    geom_bar(position="fill", stat="identity")+
    scale_fill_manual(values=MIC_profile_colours)+
    theme_minimal()+
  scale_y_continuous(labels = scales::percent)+
  labs(x="Year", y="Percentage of genomes", fill="MIC\nprediction")

MIC_year_pathogenwatch_plot


```

``` {r}
kp_profile_year / annual_multires_plot /MIC_year_pathogenwatch_plot+
  plot_annotation(tag_levels = 'a')

ggsave(height=10, width=8, file="profile_year.pdf")
ggsave(height=10, width=8, file="profile_year.png")

```

# Consumption data

``` {r}
consumption_data <- read.delim(file = "antibiotic_consumption_by_ATC3.csv", header = TRUE, sep = ",") %>%
  filter(ATC.level.3.class=="J01M-Quinolones") %>% rename(DDD=`Antibiotic.consumption..DDD.1.000.day.`) %>% select(-`ATC.level.3.class`)

```

# country means
``` {r}
# predicted R by country
PW_country_R <- PW %>% filter(species=="Klebsiella pneumoniae") %>% group_by(Country) %>% summarise(R=mean(predR=="R"))
PW_country_NWT <- PW %>% filter(species=="Klebsiella pneumoniae") %>% group_by(Country) %>% summarise(NWT=mean(predNWT=="NWT"))

# mean QRDR by country
PW_country_qrdr <- PW %>% filter(species=="Klebsiella pneumoniae") %>% group_by(Country) %>% summarise(`mean QRDR`=mean(QRDR_mutations)) 

# mean PMQR by country
PW_country_pmqr <- PW %>% filter(species=="Klebsiella pneumoniae") %>% group_by(Country) %>% summarise(`mean PMQR`=mean(acquired_genes)) 

# mean DDD by country
consumption_data_mean <- consumption_data %>% group_by(Location) %>% 
  summarise(`mean DDD`=mean(DDD)) %>% rename(Country=Location)

country_summary <- full_join(PW_country_R, PW_country_NWT) %>% 
  full_join(PW_country_qrdr) %>% full_join(PW_country_pmqr) %>% 
  full_join(PW_country_count) %>%
  full_join(consumption_data_mean)

```

``` {r}
# PMQR vs consumption
pmqr_plot <- country_summary %>% filter(countryN>=100) %>%
  ggplot(aes(x=`mean DDD`,y=`mean PMQR`))+
  geom_point()+
  stat_smooth(method="lm", color="#4781C0")+
  theme_minimal()+
  stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~")), 
    label.x = 0, color = "#4781C0", geom = "label") +
  ylim(0,1.5) +
  labs(x="Quinolone consumption (mean annual DDD)", y="Mean PMQR gene count")

# QRDR vs consumption
qrdr_plot <- country_summary %>% filter(countryN>=100) %>%
  ggplot(aes(x=`mean DDD`,y=`mean QRDR`))+
  geom_point()+
  stat_smooth(method="lm", color="#4781C0")+
  theme_minimal()+
  stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~")), 
    label.x = 0, color = "#4781C0", geom = "label") +
  ylim(0,4) +
  labs(x="Quinolone consumption (mean annual DDD)", y="Mean QRDR mutation count")

# R vs consumption
R_plot <- country_summary %>% filter(countryN>=100) %>%
  ggplot(aes(x=`mean DDD`,R))+
  geom_point()+
  stat_smooth(method="lm", color="#4781C0")+
  theme_minimal()+
  stat_cor(aes(label = paste(..rr.label.., ..p.label.., sep = "~`,`~")), 
    label.x = 0, color = "#4781C0", geom = "label") + 
  ylim(0,1.5) +
  labs(x="Quinolone consumption (mean annual DDD)", y="Predicted ciprofloxacin R")

```

``` {r}
R_plot + ggtitle("a) %R (predicted)") +  qrdr_plot + ggtitle("b) QRDR count") + pmqr_plot + ggtitle("c) PMQR count") + plot_layout(axes="collect")

ggsave(height=3, width=8, file="consumption.pdf")
ggsave(height=3, width=8, file="consumption.png")

```


``` {r}

kp_profile_country <- PW %>% filter(species=="Klebsiella pneumoniae") %>% 
  group_by(Country, rules_category) %>% count() %>%
  filter(Country %in% PW_country_count$Country[PW_country_count$countryN>500]) %>%
  left_join(PW_country_count) %>%
  ggplot(aes(x=Country, y=n, fill=as.character(rules_category))) + 
  geom_bar(stat="identity", position="fill") + 
  coord_flip() +
  scale_fill_manual(values=profile_colours, labels=profile_labels) + 
  theme_bw() + 
  scale_y_continuous(labels = scales::percent)+
  labs(x="Country", y="Percentage of genomes") + 
  theme(legend.position="none") +
  geom_text(aes(y=1.05, label = paste0("N=", countryN)))

```


```
